# De novo assembly

Using the data corrected with `trimmomatic` & `Quake` ([see Starting](https://github.com/caro46/Hymenochirus/blob/master/Starting.Rmd)) 

## Abyss

To unload one module:
```
module unload intel mkl openmpi
```
To load other modules:
```
module load gcc/4.9.2
module load openmpi/gcc-4.9.2/std/1.8.7
module load boost/gcc492-openmpi187/1.59.0
```
```
export PATH=/work/ben/abyss/bin:$PATH
abyss-pe np=8 name=BJE3814 k=64 in='BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz' 
```
Comments:

-Abyss for BJE3814 is finished: prepare the reference genome xenTro7 + BJE3814 as a reference for BJE3815. 

### Having a chimeric Abyss genome
```
abyss-pe np=8 k=64 name=BJEchimere lib='peFem peMale' peFem='/work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz' peMale='/work/ben/2016_Hymenochirus/BJE3815/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3815/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz'
```
## Abyss results statistics
```
BJE3814-stats.tab
n	n:500	L50	min	N80	N50	N20	E-size	max	sum	name
10.8e6	747074	172495	500	1089	2172	4093	2775	39687	1.244e9	BJE3814-unitigs.fa
7854121	553534	107105	500	1959	4315	8408	5555	62090	1.569e9	BJE3814-contigs.fa
7557165	403094	69676	500	2834	6580	13269	8661	95667	1.595e9	BJE3814-scaffolds.fa

more ../BJE3815/BJE3815-stats.tab
n	n:500	L50	min	N80	N50	N20	E-size	max	sum	name
10.71e6	746612	172328	500	1090	2173	4099	2775	29308	1.244e9	BJE3815-unitigs.fa
7773771	554113	107254	500	1956	4302	8412	5536	58081	1.569e9	BJE3815-contigs.fa
7481189	404914	70035	500	2815	6536	13193	8619	95232	1.593e9	BJE3815-scaffolds.fa
```
##To blast abyss outputs
On iqaluk: blast version `2.2.28+`.
Command to use after abyss:
```
module load blast/2.2.28+
```
## Format of blast output
|QueryID | SubjectID | % id | alignment length | mistmatches | gap openings | q.start | q.end | s.start | s.end | e-value | bit score| 
|---|---|---|---|---|---|---|---|---|---|---|---| 


###Prepare the reference
Need to `gunzip` the reference genome before making it blastable.
```
makeblastdb -in /work/ben/2016_Hymenochirus/xenTro7/xenTro7.fa.masked.gz -dbtype nucl -title UCSC_xenTro7 -out /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable
```
BJE3814 for BJE3815
```
makeblastdb -in /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -dbtype nucl -title Hymeno_Female_BJE3814 -out /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable
```
###Blast the output of abyss (contigs) on *X. tropicalis* genome to identify putative sex-determining region ([female reference genome for *X. tropicalis*](http://www.ncbi.nlm.nih.gov/biosample/SAMN00000117)) 
```
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -db /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTro7 -outfmt 6 -max_target_seqs 1
```
Check if contigs of our female (BJE3814) map to a region where our male (BJE3815) doesn't map.
Blast male de novo assambly against female and the opposite to find contigs only male or only female. Blast them against the female *X. tropicalis*.

On xenTrop9 (chromosomes):
```
makeblastdb -in /work/ben/2016_Hymenochirus/xenTro9/Xtropicalis_v9_repeatMasked.fa -dbtype nucl -title xenBase_xenTro9 -out /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3815/BJE3815-8.fa -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_xenTro9 -outfmt 6 -max_target_seqs 1
```
On the other sex:
```
blastn -evalue 1e-20 -query /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/BJE3814-8_BJE3815_evalue20 -outfmt 6 -max_target_seqs 1
```

To know how many contigs blast (in R, example for BJE3814 against xentrop9):
```R
blast_BJE3814_trop9 <- read.table("BJE3814-8_xenTro9",h=F,sep="\t")
scaffoldsID <- blast_BJE3814_trop9[,1]
head(scaffoldsID)
#[1] 13964 13970 13970 13970 13970 13970
scaffoldsID_unique <- unique(scaffoldsID)
head(scaffoldsID_unique)
#[1] 13964 13970 17295 18041 20125 28074
number_scaffolds_match_BJE3814_XenTrop9 <- length(scaffoldsID_unique)
number_scaffolds_match_BJE3814_XenTrop9
#[1] 38797
```
In `BJE3814-stats.tab` from Abyss output: total number of scaffolds: 7557165 -> 38797/7557165 = 0.5% (FEMALE against xenTrop9). For BJE3815 (MALE): 38928/7481189 = 0.5%
BJE3815 blast against BJE3814 (e-60): 1446606/7481189 = 19%
BJE3814 blast against BJE3815 (e-60): 1442231/7557165 = 19%

- **For e-20**:

BJE3814 blast against xenTrop9: 167698/7557165 = 2%

BJE3815 blast against xenTrop9: 165767/7481189 = 2%
```R
blast_BJE3814_BJE3815 <- read.table("BJE3814-8_BJE3815_evalue20",h=F,sep="\t")
scaffoldsID <- blast_BJE3814_BJE3815 [,1]
head(scaffoldsID)
#[1]  0  5  7  9 15 22
scaffoldsID_unique <- unique(scaffoldsID)
head(scaffoldsID_unique)
#[1]  0  5  7  9 15 22
number_scaffolds_match_BJE3814_BJE3815_e20 <- length(scaffoldsID_unique)
number_scaffolds_match_BJE3814_BJE3815_e20
#[1] 6929743
```
BJE3815 blast against BJE3814: 6866196/7481189 = 91.8%

BJE3814 blast against BJE3815: 6929743/7557165 = 91.7%

- Having sequences of scaffolds from female not mathing on male

```R
library(seqinr)
fastafile <- read.fasta(file = "BJE3814-8.fa", 
      seqtype = "DNA",as.string = TRUE, set.attributes = FALSE)
blast_res <- read.table("BJE3814-8_BJE3815-8",h=F,sep="\t")
names(fastafile)
head(blast_res[,2])

##select the contigs from the female 
# that doesn't match when blast against male

#sup_fem_caract <- fastafile[c(which(!(names(fastafile) %in% blast_res[,2])))]
sup_fem_caract <- fastafile[!(names(fastafile) %in% blast_res[,2])]
write.fasta(sequences =sup_fem_caract, names=names(sup_fem_caract),file.out = "contigs_sup_fem_caract.fa")

## After blast the contigs against xenTrop9 (female) to check female specificity
```
```
blastn -evalue 1e-20 -query /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/contigs_sup_fem_caract.fa -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814_scaffolds_sup_caract_fem_xenTro9 -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-10 -query /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/contigs_sup_fem_caract.fa -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814_scaffolds_sup_caract_fem_xenTro9_e10 -outfmt 6 -max_target_seqs 1
```
##Blast Sox3 of *X. tropicalis* against both *de novo* assambly
[cDNA & coding region](http://useast.ensembl.org/Xenopus_tropicalis/Gene/Summary?db=core;g=ENSXETG00000007607;r=GL172698.1:1304016-1305642;t=ENSXETT00000016558) (coding region tend to be more conserved) and total gene because good match (NCBI Reference Sequence: NW_004668241.1 - XenTrop7)

```
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_Sox3_cDNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Sox3_cDNA -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis7_Sox3_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Sox3_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_ar_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_ar_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_dmrta1_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_dmrta1_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_dmrta2_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_dmrta2_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Bufo_dmrt1_genebank.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Bufo_dmrt1 -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_Cyp17_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_xenTro_Cyp17_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/DMW_laevis_exon4.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_laevis_DMW_e4 -outfmt 6 -max_target_seqs 1
```

|Genes (if not specified: X. trop)|Sox3 | Sox9 |AR | Dmrt1-bufo | dmrta1 | dmrta2 | Cyp17 | Cyp19 | foxl2 | DMW-e2-laevis | DMW-e3-laevis | DMW-e4-laevis | nop2|got1|mat1a|pgam|SEC23IP|zranb1|tubgcp2|AMHR2|SMARCB1|FGA|CHD1|MAP1B|VLDLR|KANK1|dock8|RAD23B|fmr1|NLGN4X|
|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
|Female | &#10004;|&#10004;|&#10004;|&#88;|&#88;|&#88;|&#10004;|e-10|&#10004;|&#88;|&#88;|&#88;|&#88;|&#88;|e-10|&#88;|e-10|e-10|e-10|&#88;|e-10|e-10|e-10|e-10|e-10|e-10|e-10|e-10|e-10|e-10|
|scaffoldIDabyss | 25157710|24992484|25169781|&#88;|&#88;|&#88;|25206211|25193799|25186476|&#88;|&#88;|&#88;|&#88;|&#88;|24956237|&#88;|25123958|24541643|25234276|&#88;|24649470|25116792|25231621|25163654|25230157|25136231|24638043|25190621|24913548|24599310|
|Male |&#10004;|&#10004;|&#10004;|&#88;|&#88;|&#88;|&#10004;|e-10|&#10004;|&#88;|&#88;|&#88;|&#88;|&#88;|e-10|&#88;|e-10|e-10|e-10|&#88;|e-10|e-10|e-10|e-10|e-10|e-10|e-10|e-10|e-10|e-10|
|scaffoldIDabyss | 24881349|24702845|24835785|&#88;|&#88;|&#88;|24853046|24128721|24861975|&#88;|&#88;|&#88;|&#88;|&#88;|24900110|&#88;|24884876|24268182|24946651|&#88;|24348384|24882348|24958695|24923528|24908458|24810213|24809088|24433179|24928036|24896648|

(Try with `-evalue 1e-10` for dmrta1-bufo/-trop/dmrta2/Cyp19/DMW/nop2: Cyp19 in female+male). Got1/AMHR2 only e-10/e-2.

See [Uno et al. 2008](http://www.ncbi.nlm.nih.gov/pubmed/18484182) for more sex-determining genes:  AR, SF-1/Ad4BP, Sox3, Dmrt1 (might not related to sex determination in *Rana rugosa*). And [Nakamura 2009](http://www.ncbi.nlm.nih.gov/pubmed/18996493): CYP19, CYP17, Foxl2. See also [Uno et al. 2013] (http://www.nature.com/hdy/journal/v111/n5/full/hdy201365a.html) and [Brelsford et al. 2013](http://onlinelibrary.wiley.com/doi/10.1111/evo.12151/pdf).

> Compare size + identity of the blast results between male/female.
> We checked for difference in coverage between male/female of the genes (expect: 2 copies in homogametic sex / 1 in homogametic sex -> expect diff. in cov by ~ *2 if sex-gene linked). No one appears after the use of the perl script > check if because of scaffold size or mean cov <*2... 

## Stampy (because Cortex not working for the moment)
```
/work/ben/stampy-1.0.28/stampy.py -t8 -g /work/ben/2016_Hymenochirus/xenTro7/xenTro7 -h /work/ben/2016_Hymenochirus/xenTro7/xenTro7 --substitutionrate=0.10 -o /work/ben/2016_Hymenochirus/BJE3814/BJE3814_stampy_xenTro7_paired.sam -M /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz
```
Estimate coverage among the reference genome (samtools mpileup) + plot with `R`.

```
/work/ben/macaque_RAD_TAGs/samtools-0.1.18/samtools flagstat BJE3815_stampy_xenTro7_paired_sorted.bam > BJE3815_stampy_xenTro7_paired_sorted_flagstat.txt
```
Output from flagstat from BJE3815
```
87370213 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
5278107 + 0 mapped (6.04%:-nan%)
87370213 + 0 paired in sequencing
43685107 + 0 read1
43685106 + 0 read2
1261700 + 0 properly paired (1.44%:-nan%)
2290444 + 0 with itself and mate mapped
2987663 + 0 singletons (3.42%:-nan%)
793302 + 0 with mate mapped to a different chr
121082 + 0 with mate mapped to a different chr (mapQ>=5)
```
Output from flagstat from BJE3814
```
102080926 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
6247778 + 0 mapped (6.12%:-nan%)
102080926 + 0 paired in sequencing
51040463 + 0 read1
51040463 + 0 read2
1502320 + 0 properly paired (1.47%:-nan%)
2757848 + 0 with itself and mate mapped
3489930 + 0 singletons (3.42%:-nan%)
969546 + 0 with mate mapped to a different chr
144536 + 0 with mate mapped to a different chr (mapQ>=5)
```

> Results Stampy: bad mapping -> use bwa or bowtie to map male & female reads against Abyss genome (of both sex) and infer the coverage using Samtools > use bwa mem command "For 70bp or longer Illumina, 454, Ion Torrent and Sanger reads, assembly contigs and BAC sequences, BWA-MEM is usually the preferred algorithm."  from [bwa website](http://bio-bwa.sourceforge.net/)

```perl
#!/usr/bin/perl                                                                                                                                                   

use warnings;
use strict;

# This script will execute alignment functions for paired end reads
# using bwa mem for reference contigs

my $path_to_bwa="/work/ben/macaque_RAD_TAGs/bwa-0.6.2";
my $path_to_samtools="/work/ben/macaque_RAD_TAGs/samtools-0.1.18";
my $path_to_data="/work/ben/2016_Hymenochirus/BJE3814";
my $path_to_genome="/work/ben/2016_Hymenochirus/BJE3815";
my $genome="BJE3815-8.fa";
my $individual=" BJE3814";

my $commandline;
my $status;

# Index
$commandline = $path_to_bwa."\/bwa index -a bwtsw ".$path_to_genome."\/".$genome;

$status = system($commandline);

# bwa mem
$commandline = $path_to_bwa."\/bwa mem ".$path_to_genome. "\/".$genome. $path_to_data."\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.gz".$path_to_data."\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.sam";
$status = system($commandline);

# Samtools : sam to bam
$commandline=$path_to_samtools."\/samtools view -bt ".$path_to_genome."\/".$genome." -o ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.bam ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.sam";
$status = system($commandline);
$commandline=$path_to_samtools."\/samtools sort ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.bam ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815_sorted";
$status = system($commandline);
$commandline= $path_to_samtools."\/samtools index ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815_sorted.bam";
$status = system($commandline);
```
# SOAP de novo 
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s /work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/hymeno.config -K 64 -R -o SOAP_Hymeno_genome 1>ass.log 2>ass.err

/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s /work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/hymenoFEMALE.config -K 64 -R -o SOAP_Hymeno_FEMALEgenome 1>ass.log 2>ass.err
```
## SOAP chimeric genome (reads from mom + dad) - statistics for the scaffolds
```
Size_includeN	2818397698
Size_withoutN	2699755794
Scaffold_Num	9612629
Mean_Size	293
Median_Size	130
Longest_Seq	116517
Shortest_Seq	100
Singleton_Num	9021371
Average_length_of_break(N)_in_scaffold	12

scaffolds>100 	9426455	98.06%
scaffolds>500 	769632	8.01%
scaffolds>1K  	386272	4.02%
scaffolds>10K 	16535	0.17%
scaffolds>100K	3	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	832817850	29.55%
Nucleotide_C	538677655	19.11%
Nucleotide_G	521887058	18.52%
Nucleotide_T	806373231	28.61%
GapContent_N	118641904	4.21%
Non_ACGTN	0	0.00%
GC_Content	39.28%		(G+C)/(A+C+G+T)

N10	9297	19312
N20	4885	62312
N30	2624	141850
N40	1294	297103
N50	631	615567
N60	292	1285271
N70	161	2606172
N80	130	4703979
N90	116	6991582
```
##Female SOAP genome
```
<-- Information for assembly Scaffold 'SOAP_Hymeno_FEMALEgenome.scafSeq'.(cut_off_length < 100bp) -->

Size_includeN	2454177557
Size_withoutN	2346198294
Scaffold_Num	6187274
Mean_Size	396
Median_Size	131
Longest_Seq	150321
Shortest_Seq	100
Singleton_Num	5677197
Average_length_of_break(N)_in_scaffold	17

Known_genome_size	NaN
Total_scaffold_length_as_percentage_of_known_genome_size	NaN

scaffolds>100 	6078816	98.25%
scaffolds>500 	610986	9.87%
scaffolds>1K  	344300	5.56%
scaffolds>10K 	29240	0.47%
scaffolds>100K	5	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	721929527	29.42%
Nucleotide_C	465577499	18.97%
Nucleotide_G	453924352	18.50%
Nucleotide_T	704766916	28.72%
GapContent_N	107979263	4.40%
Non_ACGTN	0	0.00%
GC_Content	39.19%		(G+C)/(A+C+G+T)

N10	17162	9720
N20	10089	28795
N30	6216	60061
N40	3666	111708
N50	1886	205294
N60	817	406805
N70	321	902622
N80	138	2090865
N90	123	3983586
```
##Male SOAP genome
```
<-- Information for assembly Scaffold 'SOAP_Hymeno_MALEgenome.scafSeq'.(cut_off_length < 100bp) -->

Size_includeN	2448945709
Size_withoutN	2340724728
Scaffold_Num	6166746
Mean_Size	397
Median_Size	131
Longest_Seq	132924
Shortest_Seq	100
Singleton_Num	5653637
Average_length_of_break(N)_in_scaffold	17

Known_genome_size	NaN
Total_scaffold_length_as_percentage_of_known_genome_size	NaN

scaffolds>100 	6060456	98.28%
scaffolds>500 	617965	10.02%
scaffolds>1K  	346019	5.61%
scaffolds>10K 	28498	0.46%
scaffolds>100K	5	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	720913996	29.44%
Nucleotide_C	463811828	18.94%
Nucleotide_G	452376524	18.47%
Nucleotide_T	703622380	28.73%
GapContent_N	108220981	4.42%
Non_ACGTN	0	0.00%
GC_Content	39.14%		(G+C)/(A+C+G+T)

N10	16939	9730
N20	9887	29079
N30	6090	60921
N40	3581	113624
N50	1835	209537
N60	805	415005
N70	321	911704
N80	140	2087590
N90	124	3972006
```

# Mapping with `bwa` 

Ref genome = `X. tropicalis v9.0 genome assembly` from [XenBase](http://www.xenbase.org/other/static/ftpDatafiles.jsp)
```perl
#!/usr/bin/perl                                                                                                             

use warnings;
use strict;

# This script will execute alignment functions for paired end reads                                                          
# using bwa mem for reference contigs                                                                                        

my $path_to_bwa="/work/ben/macaque_RAD_TAGs/bwa-0.6.2";
my $path_to_samtools="/work/ben/macaque_RAD_TAGs/samtools-0.1.18";
my $path_to_data="/work/ben/2016_Hymenochirus/BJE3814";
my $path_to_genome="/work/ben/2016_Hymenochirus/xenTro9";
my $genome="Xtropicalis_v9_repeatMasked.fa";
my $individual="BJE3814";

my $commandline;
my $status;

# Index                                                                                                                      
#$commandline = $path_to_bwa."\/bwa index -a bwtsw ".$path_to_genome."\/".$genome;                                           

#$status = system($commandline);                                                                                             

#bwa sampe                                                                                                                   
$commandline = $path_to_bwa."\/bwa aln ".$path_to_genome."\/".$genome." ".$path_to_data."\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.sai";
$status = system($commandline);
$commandline = $path_to_bwa."\/bwa aln ".$path_to_genome."\/".$genome." ".$path_to_data."\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.sai";
$status = system($commandline);

# bwa sampe                                                                                                                  
$commandline = $path_to_bwa."\/bwa sampe ".$path_to_genome. "\/".$genome." ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.sai ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.sai ".$path_to_data."\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.gz ".$path_to_data."\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9.sam";
$status = system($commandline);

# Samtools : sam to bam                                                                                                      
$commandline=$path_to_samtools."\/samtools view -bt ".$path_to_genome."\/".$genome." -o ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_XenTro9.bam ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9.sam";
$status = system($commandline);
$commandline=$path_to_samtools."\/samtools sort ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9.bam ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9_sorted";
$status = system($commandline);
$commandline= $path_to_samtools."\/samtools index ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9_sorted.bam";
$status = system($commandline);
```
Map BJE3814 & BJE3815 to xenTrop9 and then merge .bam files and use `mpileup``bcftools``vcfutils` to have a consensus sequence
If need to add RG tags:
```
perl -e 'print "@RG\\tID:BJE3814\\tSM:hs\\tLB:BJE3814\\tPL:Illumina\\n@RG\\tID:BJE3815\\tSM:hs\\tLB:BJE3815\\tPL:Illumina\\n"' > rg.txt
samtools merge -rh rg.txt merged.bam BJE3814.bam BJE3815.bam
```
`samtools merge out.bam in1.bam in2.bam in3.bam`

`samtools mpileup [-EBugp] [-C capQcoef] [-r reg] [-f in.fa] [-l list] [-Q minBaseQ] [-q minMapQ] in.bam [in2.bam [...]]`

To have a the consensus sequence:

`samtools mpileup -uf ref.fa aln.bam | bcftools call -c | vcfutils.pl vcf2fq > cns.fq`

If mapping ok with xenTrop9 as a reference genome: use it to calculate the nucleotide diversity, otherwise use the chimere (female + male) abyss assembly. 

To do when Iqaluk OK or on info
Identify repeated sequences using `Repeatmasker` on the 2 de novo assembly (female + male) ([Burger & Palmieri 2013](http://jhered.oxfordjournals.org/content/105/6/933.full) an example for using Repeatmasker   
Compare with repeat elements identified in [X. trop](http://www.repeatmasker.org/genomicDatasets/RMGenomicDatasets.html)

- Reads mapping against XenTrop9 genome
```
bwa mem /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa>aln_bwa_mem_trop9_BJE3814.sam

samtools view -bt /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa -o /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814.bam /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814.sam

samtools sort /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814.bam /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814_sorted

samtools index /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814_sorted.bam

samtools flagstat /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814_sorted.bam > aln_bwa_mem_trop9_BJE3814_sorted_flagstat.txt
```
- Reads mapping against their own abyss genome
```
bwa index /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_BJE3814_reads_abyss.sam

samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_BJE3814_reads_abyss.sam

samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss_sorted

samtools index /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss_sorted.bam

samtools flagstat /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss_sorted.bam > aln_bwa_mem_BJE3814_reads_abyss_sorted_flagstat.txt
```
On info bwa: Version: 0.7.8-r455

- Reads mapping against the other abyss genome (ex.: reads female against male abyss genome...)
```
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_BJE3814abyss_BJE3815reads.sam

bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3815-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_BJE3815abyss_BJE3814reads.sam

samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_BJE3814abyss_BJE3815reads.sam

samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted

samtools index /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted.bam

samtools flagstat /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted.bam > aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted_flagstat.txt
```
- Estimate the coverage for each abyss genome (female or male) using the 2 types of read (female + male)
```
samtools depth aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted.bam aln_bwa_mem_BJE3814_reads_abyss_sorted.bam > abyssBJE3814_depth_BJE3814_BJE3815.depth

samtools depth aln_bwa_mem_BJE3815abyss_BJE3814reads_sorted.bam aln_bwa_mem_BJE3815_reads_abyss_sorted.bam > abyssBJE3815_depth_BJE3815_BJE3814.depth
```
Mean by scaffold with R:
```R
cov_by_scaffold=read.table("abyssBJE3814_depth_BJE3814_BJE3815.depth",sep="\t",h=F)
cov_MALE <- aggregate(cov_by_scaffold[,3]~cov_by_scaffold[,1],FUN=mean)
cov_FEMALE <- aggregate(cov_by_scaffold[,4]~cov_by_scaffold[,1],FUN=mean)
pdf('Coverage_abyssBJE3814_readsBJE3815_BJE3814.pdf')
plot(cov_MALE[,2]~cov_MALE[,1],type="l",col="blue",ylab="depth",xlab="scaffolds",main="Depth coverage of abyss genome of BJE3814 (Hymenochirus Female)")
lines (cov_FEMALE[,2], col="pink")
legend('topright',c("Female BJE3814","Male BJE3815"),lty=c(1,1),lwd=c(2.5,2.5),col=c("pink","blue"))
dev.off()

```
File too big for R. So obtain mean by scaffold using a perl script:
```perl
#!/usr/bin/perl

use strict;
use warnings;

open my $fh, '<', "abyssBJE3814_depth_BJE3814_BJE3815.depth" or die $!;

my $new_file = "coverage_scaffold_abyssBJE3814.txt";
# Use the open() function to create the file.
unless(open FILE, '>'.$new_file) {
    # Die with error message 
    # if we can't open it.
    die "\nUnable to create $new_file\n";
}

my $covM_mean ;
my $covF_mean ;
my $totalM = 0;
my $totalF = 0;
my $count = 0;

while (<$fh>) {

    my (@scaffold, $location, $covM, $covF) = split(/\t/);
foreach @scaffold{
    $totalM += $covM;
    $totalF += $covF;
    $count += 1;
    $covM_mean = $totalM/$count ;
    $covF_mean = $totalF/$count ;
}
}

# Write some text to the new_file.

my $y;
    for ($y=0; $y<=$#scaffold; $y ++) {
print FILE $scaffold[$y],"\t",$covM_mean[$y],"\t",$covF_mean[$y],"\n" ;
}
# close the file.
close FILE;
exit 0;
```
Use R to identify which scaffolds of the female has no coverage in male but coverage in female
```R
######################FEMALE abyss

setwd("/Users/Ben/Hymenochirus_genome")
depth_fem_abyss <- read.table("output_abyssBJE3814_depth_BJE3814_BJE3815",h=T,sep="\t")
head(depth_fem_abyss)
plot(depth_fem_abyss$femalecoverage~depth_fem_abyss$Scaffold, col="pink",type="l")
lines(depth_fem_abyss$malecoverage,col="blue")

##ordering the table to have all the scaffolds with no coverage in male 1st
DFA_order <- depth_fem_abyss[order(depth_fem_abyss$femalecoverage,depth_fem_abyss$malecoverage),]
DFA_order_m <- depth_fem_abyss[order(depth_fem_abyss$malecoverage,depth_fem_abyss$femalecoverage),]

head(DFA_order)
head(DFA_order_m)
plot(DFA_order_m$femalecoverage~DFA_order_m$Scaffold, col="pink",type="l")
lines(DFA_order_m$malecoverage,col="blue")
depth_fem_abyss$depth_fem_abyss <- as.numeric(depth_fem_abyss$depth_fem_abyss)

max(DFA_order_m$femalecoverage) #561.7466
min(DFA_order_m$femalecoverage) #9.801199
mean(DFA_order_m$femalecoverage) #31.46771

max(DFA_order_m$malecoverage) #926.2464
min(DFA_order_m$malecoverage) #0
mean(DFA_order_m$malecoverage) #24.59831

#boxplot(DFA_order_m$femalecoverage)
##to plot only wher we have no coverage in male
DFA_subset <- DFA_order_m[DFA_order_m$malecoverage==0,]
head(DFA_subset)
x11()
plot(DFA_subset$femalecoverage~DFA_subset$Scaffold,type="o",col="pink")
tail(DFA_subset)
#lines(DFA_subset$femalecoverage,col="pink")
#lines(DFA_subset$malecoverage,col="blue")
DFA_order_m$femalecoverage<- as.numeric(DFA_order_m$femalecoverage)
DFA_subset2 <- DFA_subset[DFA_order_m$femalecoverage>200,]
head(DFA_subset2)

max(DFA_subset$femalecoverage) #60.39382
min(DFA_subset$femalecoverage) #15.561
mean(DFA_subset$femalecoverage) #26.2403
length(DFA_subset$femalecoverage) #51
min(DFA_subset$bp) #1015
boxplot(DFA_subset$bp) 
hist(DFA_subset$bp)
max(DFA_subset$bp) #5585
mean(DFA_subset$bp) #1798.294

######################Preparing for blasting specific scaffolds to X. tropicalis female
library(seqinr)
fastafile <- read.fasta(file = "BJE3814-8.fa", 
                        seqtype = "DNA",as.string = TRUE, set.attributes = FALSE)
##need to delete "scaffold" from the name of the sequences
DFA_subset$Scaffold <- gsub('scaffold_','',DFA_subset$Scaffold)
sup_fem_caract_depth <- fastafile[(names(fastafile) %in% DFA_subset[,1])]
write.fasta(sequences =sup_fem_caract_depth, names=names(sup_fem_caract_depth),file.out = "scaffolds_sup_fem_caract_depth.fa")
head(sup_fem_caract_depth)
```
We blast scaffolds potentially female against xentrop9 on Iqaluk first. Not good match.
When we blast the biggest "female specific" scaffold (24634183) identified with the coverage (cov. 0 for male) against: *X. tropicalis* 9 on xenbase: no match, 7 on Ensembl: multiple hit / length 24-58 bp / e-value ~9.10^-3 1.5^-6.
We also identified scaffolds of female when we have a mean of scaffold coverage for male: `round(DFA_order_m$malecoverage)<=0`(xenbase: see Ensembl res., Ensembl: no hit with length >50bp for 16475015 which appears matching multiple times against Chr.07 when we blast localy) & `round(DFA_order_m$malecoverage)<=1`.

Results of blast when blasting specific female scaffolds identified by the coverage (e-value 10^-5)
Example of a command:

`blastn -evalue 1e-5 -query /work/ben/2016_Hymenochirus/BJE3814/blast/scaffolds_sup_fem_caract_depth_round0.fa -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814_scaffolds_sup_caract_fem_depth_round0_results_xenTro9_e5 -outfmt 6 -max_target_seqs 1`

```
[ben@iqaluk:/work/ben/2016_Hymenochirus/BJE3814/blast] more BJE3814_scaffolds_sup_caract_fem_depthresults_xenTro9_e5
14152388	Chr06	84.85	66	10	0	769	834	32448736	32448801	7e-09	67.6
17520648	scaffold_6869	82.67	75	11	2	705	778	3612	3685	3e-08	65.8
24809196	Chr08	95.92	49	2	0	1947	1995	38166875	38166923	2e-12	80.5
24809196	Chr08	97.62	42	1	0	1954	1995	16236610	16236651	3e-10	73.1
24809196	Chr08	95.45	44	2	0	1952	1995	29811484	29811527	1e-09	71.3
24809196	Chr08	95.45	44	2	0	1952	1995	74220364	74220321	1e-09	71.3
24809196	Chr08	95.35	43	2	0	1953	1995	41087090	41087048	4e-09	69.4
24809196	Chr08	95.35	43	2	0	1953	1995	91538871	91538913	4e-09	69.4
24809196	Chr08	100.00	36	0	0	1953	1988	99772065	99772030	1e-08	67.6
24809196	Chr08	94.74	38	2	0	1952	1989	48886695	48886732	2e-06	60.2
24973669	scaffold_69	89.06	64	7	0	1617	1680	329682	329619	1e-12	80.5
25029956	Chr06	94.74	38	2	0	1058	1095	121909434	121909397	2e-06	60.2
25067851	Chr03	100.00	32	0	0	10	41	26199677	26199646	3e-06	60.2
```
For **FEMALE**:

-**24761011** seemed to be a good scaffold (length of 2439bp, map with lenght <100bp, multiple hits on Chr.07) but when blast on Ensembl and Xenbase map ~also not bad on other chr. -> repeatmasker: nothing identified as repeat elements...)

-**24680304**: 1353 bp, 260 bp identified as RE, Chr.05/02/09/01/04

-**24579464**: chr.10, also 03/04/02: when reblast on web and 07 -> repeatmasker: nothing

-**25155228** chr.07 - 9002 bp - repeatmasker 332 bp masked (1 LINE L2/CR1/Rex: 273 bp) 679 map.Position 53096997-53096331 -> arm  

INTERESTING:

Female: scaffold_25155228: only BJE3814_scaffolds_sup_caract_fem_depth_round1_results_xenTro9_e5 longer than 500bp. Blast against male scaff24388608. Don't find it in the mean cov per scaffold against abyss male. Use grep command to have cov for all the positions of this scaffold.
```
grep '24388608' abyssBJE3815_depth_BJE3815_BJE3814.depth> abyssBJE3815_scaff24388608_depth_BJE3815_BJE3814.depth
```
```
blastn -evalue 1e-20 -query /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/BJE3814_scaff25155228.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/BJE3814-scaff25155228_BJE3815_evalue20 -outfmt 6 -max_target_seqs 1
```
Blast results (same with e-value -5)
```
25155228	24388608	82.66	1113	145	32	3819	4911	3456	2372	0.0	 942
25155228	24388608	77.89	615	82	28	3213	3809	5317	4739	8e-88	 333
```
```
blastn -evalue 1e-20 -query /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/BJE3814_scaff25155228.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/BJE3814-scaff25155228_BJE3814_evalue20 -outfmt 6 -max_target_seqs 1
```
Blast only against itself...
Blast Male scaff24388608 against itself and female.
Let's align scaffold female scaffold_25155228 and male scaff24388608.
```
blastn -evalue 1e-5 -query /work/ben/2016_Hymenochirus/mafft_align/BJE3815_scaff24388608.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast_against_BJE3814/BJE3815-scaff24388608_BJE3814_evalue5 -outfmt 6 -max_target_seqs 1
```
```
Query_1	24615454	99.97	6467	2	0	1	6467	6467	1	0.0	1.193e+04
Query_1	24615454	83.98	181	21	6	55	231	6237	6413	6e-38	 167
```
Blast only against itself (24388608)...
RepeatMaker on male :L2/CR1/Rex 1 514 bp 7.95 %
MAFFT alignment: 

- Reads mapping against the chimeric SOAP de novo genome
```
bwa index /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_ChimereSOAP1415_BJE3814reads.sam

samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3814reads.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_ChimereSOAP1415_BJE3814reads.sam

samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3814reads.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3814reads_sorted

TO RUN when others finished:
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_ChimereSOAP1415_BJE3815reads.sam

samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3815reads.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_ChimereSOAP1415_BJE3815reads.sam
samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3815reads.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3815reads_sorted
samtools index /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3815reads_sorted.bam
samtools flagstat /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3815reads_sorted.bam > aln_bwa_mem_ChimereSOAP1415_BJE3815reads_sorted_flagstat.txt

samtools depth aln_bwa_mem_ChimereSOAP1415_BJE3814reads_sorted.bam aln_bwa_mem_ChimereSOAP1415_BJE3815reads_sorted.bam > SOAPchimereBJE3814_BJE3815_depth_reads_BJE3814_BJE3815.depth
```
- Reads against SOAP female & male genomes

Female genome:
```
bwa index /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_FEMALEgenome.scafSeq
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_FEMALEgenome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads.sam
samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_FEMALEgenome.scafSeq -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads.sam
samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads_sorted
samtools index /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads_sorted.bam
samtools flagstat /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads_sorted.bam > aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads_sorted_flagstat.txt
```
Results
```
evanslab:map_reads_to_abyss_genomes % more aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3814reads_sorted_flagstat.txt
976329475 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
976187730 + 0 mapped (99.99%:-nan%)
976329475 + 0 paired in sequencing
491308989 + 0 read1
485020486 + 0 read2
570664029 + 0 properly paired (58.45%:-nan%)
976026768 + 0 with itself and mate mapped
160962 + 0 singletons (0.02%:-nan%)
438986696 + 0 with mate mapped to a different chr
282020843 + 0 with mate mapped to a different chr (mapQ>=5)
```
```
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_FEMALEgenome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz > /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_SOAP_Hymeno_FEMALEgenome_BJE3815reads.sam
```
Male genome:
```
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_MALEgenome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz > /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads.sam
samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_MALEgenome.scafSeq -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads.sam
samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads_sorted
samtools index /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads_sorted.bam
samtools flagstat /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads_sorted.bam > aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads_sorted_flagstat.txt
```
Results
```
evanslab:map_reads_to_abyss_genomes % screen -r index_SOAP_male
[detached]
evanslab:map_reads_to_abyss_genomes % more aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3815reads_sorted_flagstat.txt
992443965 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
992271327 + 0 mapped (99.98%:-nan%)
992443965 + 0 paired in sequencing
498711108 + 0 read1
493732857 + 0 read2
581245873 + 0 properly paired (58.57%:-nan%)
992097576 + 0 with itself and mate mapped
173751 + 0 singletons (0.02%:-nan%)
444491490 + 0 with mate mapped to a different chr
285074221 + 0 with mate mapped to a different chr (mapQ>=5)
```
```
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_MALEgenome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_SOAP_Hymeno_MALEgenome_BJE3814reads.sam
```
- Blast SOAP scaffolds "female specific" (coverage) =>10,000bp
```
blastn -evalue 1e-10 -query /work/ben/2016_Hymenochirus/SOAP_genomes/SOAPchim_scaffolds_sup10000bp_sup_FEMALE_BJE3814_caract_depth_round1.fa -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/SOAP_genomes/SOAPchim_scaffolds_sup10000bp_sup_FEMALE_BJE3814_caract_depth_round1_results_xenTro9_e10 -outfmt 6 -max_target_seqs 1
```
Blast of them against Abyss - scaffolds "female specific" (coverage reads F & M against AbyssF)
```
makeblastdb -in /work/ben/2016_Hymenochirus/BJE3814/blast/scaffolds_sup_fem_caract_depth_round1.fa -dbtype nucl -title Hymeno_Female_BJE3814_scaffolds_sup_fem_caract_depth_round1 -out /work/ben/2016_Hymenochirus/SOAP_genomes/blast_spec_cov/BJE3814_abyss_depth_round1_scafSupCaract_blastable
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/SOAP_genomes/SOAPchim_scaffolds_sup10000bp_sup_FEMALE_BJE3814_caract_depth_round1.fa -db /work/ben/2016_Hymenochirus/SOAP_genomes/blast_spec_cov/BJE3814_abyss_depth_round1_scafSupCaract_blastable -out /work/ben/2016_Hymenochirus/SOAP_genomes/blast_spec_cov/SOAPchim_scaffolds_sup10000bp_sup_FEMALE_BJE3814_caract_depth_round1_BJE3814_abyss_depth_round1_scafSupCaract -outfmt 6 -max_target_seqs 1

#Without size lim (all scaff caract)
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/SOAP_genomes/SOAPchim_scaffolds_sup_FEMALE_BJE3814_caract_depth_round1.fa -db /work/ben/2016_Hymenochirus/SOAP_genomes/blast_spec_cov/BJE3814_abyss_depth_round1_scafSupCaract_blastable -out /work/ben/2016_Hymenochirus/SOAP_genomes/blast_spec_cov/SOAPchim_scaffolds_sup_FEMALE_BJE3814_caract_depth_round1_BJE3814_abyss_depth_round1_scafSupCaract -outfmt 6 -max_target_seqs 1
```
## [MUMMER](http://mummer.sourceforge.net/manual/#description)

- Can be used for align multiple finished or draft genomes and identify at some point some repeats regions (align seq against itself)

Us: Reference (XenTrop9) divergent from the 2 other sequences (BJE3814 & BJE3815). If not working just nucmer on Abyss Assembly (BJE3814 & BJE3815) to identify similar and different regions. (ON INFO)
```
promer --prefix=ref_qry ref.fasta qry.fasta
/usr/local/mummer/MUMmer3.23/promer -o --prefix=xentrop9_BJE3814 Xtropicalis_v9_repeatMasked.fa BJE3814-8.fa 
/usr/local/mummer/MUMmer3.23/mummer -L -c Xtropicalis_v9_repeatMasked.fa BJE3814-8.fa BJE3815-8.fa

/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/nucmer -o --prefix=Abyss_BJE3814_BJE3815 /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa /home/caroline/hymeno/MUMMER_analysis/BJE3815-8.fa
/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/mummerplot /home/caroline/hymeno/MUMMER_analysis/BJE3814_3815_nucmer/Abyss_BJE3814_BJE3815.delta -R /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa -Q /home/caroline/hymeno/MUMMER_analysis/BJE3815-8.fa -S --png --prefix mummerplot_nucmer_BJE3814_BJE3815 --filter --layout

/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/promer -o --prefix=xentrop9_BJE3814 /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa
/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/mummerplot /home/caroline/hymeno/MUMMER_analysis/BJE3814_promer/xentrop9_BJE3814.delta -R /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa -Q /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa --png --prefix mummerplot_promer_xenTrop9_BJE3814 --filter --layout
```

## Ideas to try to improve *de novo* assembly:
- Use [pagit tools](http://www.sanger.ac.uk/science/tools/pagit): ABACAS & IMAGE to order the scaffolds according to a reference genome and add "N" between the scaffolds.

To check if something is running on the background:
```
ps -elf | grep 'stampy'
```
