# De novo assembly

Using the data corrected with `trimmomatic` & `Quake` ([see Starting](https://github.com/caro46/Hymenochirus/blob/master/Starting.Rmd)) 

## Abyss

To unload one module:
```
module unload intel mkl openmpi
```
To load other modules:
```
module load gcc/4.9.2
module load openmpi/gcc-4.9.2/std/1.8.7
module load boost/gcc492-openmpi187/1.59.0
```
```
export PATH=/work/ben/abyss/bin:$PATH
abyss-pe np=8 name=BJE3814 k=64 in='BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz' 
```
##To blast abyss outputs
On iqaluk: blast version `2.2.28+`.
Command to use after abyss:
```
module load blast/2.2.28+
```
###Prepare the reference
```
makeblastdb -in /work/ben/2016_Hymenochirus/xenTro7/xenTro7.fa.masked.gz -dbtype nucl -title UCSC_xenTro7 -out /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable
```
###Blast the output of abyss (contigs) on *X. tropicalis* genome to identify putative sex-determining region ([female reference genome for *X. tropicalis*](http://www.ncbi.nlm.nih.gov/biosample/SAMN00000117)) 
```
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -db /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8_xenTro7 -outfmt 6 -max_target_seqs 1
```
Check if contigs of our female (BJE3814) map to a region where our male (BJE3815) doesn't map.
Blast male de novo assambly against female and the opposite to find contigs only male or only female. Blast them against the female *X. tropicalis*.

## Stampy (because Cortex not working for the moment)
```
/work/ben/stampy-1.0.28/stampy.py -t8 -g /work/ben/2016_Hymenochirus/xenTro7/xenTro7 -h /work/ben/2016_Hymenochirus/xenTro7/xenTro7 --substitutionrate=0.10 -o /work/ben/2016_Hymenochirus/BJE3814/BJE3814_stampy_xenTro7_paired.sam -M /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz
```
