# De novo assembly

Using the data corrected with `trimmomatic` & `Quake` ([see Starting](https://github.com/caro46/Hymenochirus/blob/master/Starting.Rmd)) 

## Abyss

To unload one module:
```
module unload intel mkl openmpi
```
To load other modules:
```
module load gcc/4.9.2
module load openmpi/gcc-4.9.2/std/1.8.7
module load boost/gcc492-openmpi187/1.59.0
```
```
export PATH=/work/ben/abyss/bin:$PATH
abyss-pe np=8 name=BJE3814 k=64 in='BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz' 
```
Comments:

-Abyss for BJE3814 is finished: prepare the reference genome xenTro7 + BJE3814 as a reference for BJE3815. 

### Having a chimeric Abyss genome
```
abyss-pe np=8 k=64 name=BJEchimere lib='peFem peMale' peFem='/work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz' peMale='/work/ben/2016_Hymenochirus/BJE3815/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3815/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz'
```
## Abyss results statistics
```
BJE3814-stats.tab
n	n:500	L50	min	N80	N50	N20	E-size	max	sum	name
10.8e6	747074	172495	500	1089	2172	4093	2775	39687	1.244e9	BJE3814-unitigs.fa
7854121	553534	107105	500	1959	4315	8408	5555	62090	1.569e9	BJE3814-contigs.fa
7557165	403094	69676	500	2834	6580	13269	8661	95667	1.595e9	BJE3814-scaffolds.fa

more ../BJE3815/BJE3815-stats.tab
n	n:500	L50	min	N80	N50	N20	E-size	max	sum	name
10.71e6	746612	172328	500	1090	2173	4099	2775	29308	1.244e9	BJE3815-unitigs.fa
7773771	554113	107254	500	1956	4302	8412	5536	58081	1.569e9	BJE3815-contigs.fa
7481189	404914	70035	500	2815	6536	13193	8619	95232	1.593e9	BJE3815-scaffolds.fa
```
##To blast abyss outputs
On iqaluk: blast version `2.2.28+`.
Command to use after abyss:
```
module load blast/2.2.28+
```
## Format of blast output
|QueryID | SubjectID | % id | alignment length | mistmatches | gap openings | q.start | q.end | s.start | s.end | e-value | bit score| 
|---|---|---|---|---|---|---|---|---|---|---|---| 


###Prepare the reference
Need to `gunzip` the reference genome before making it blastable.
```
makeblastdb -in /work/ben/2016_Hymenochirus/xenTro7/xenTro7.fa.masked.gz -dbtype nucl -title UCSC_xenTro7 -out /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable
```
BJE3814 for BJE3815
```
makeblastdb -in /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -dbtype nucl -title Hymeno_Female_BJE3814 -out /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable
```
###Blast the output of abyss (contigs) on *X. tropicalis* genome to identify putative sex-determining region ([female reference genome for *X. tropicalis*](http://www.ncbi.nlm.nih.gov/biosample/SAMN00000117)) 
```
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -db /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTro7 -outfmt 6 -max_target_seqs 1
```
Check if contigs of our female (BJE3814) map to a region where our male (BJE3815) doesn't map.
Blast male de novo assambly against female and the opposite to find contigs only male or only female. Blast them against the female *X. tropicalis*.

On xenTrop9 (chromosomes):
```
makeblastdb -in /work/ben/2016_Hymenochirus/xenTro9/Xtropicalis_v9_repeatMasked.fa -dbtype nucl -title xenBase_xenTro9 -out /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3815/BJE3815-8.fa -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_xenTro9 -outfmt 6 -max_target_seqs 1
```
On the other sex:
```
blastn -evalue 1e-20 -query /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast_against_BJE3815/BJE3814-8_BJE3815_evalue20 -outfmt 6 -max_target_seqs 1
```

To know how many contigs blast (in R, example for BJE3814 against xentrop9):
```R
blast_BJE3814_trop9 <- read.table("BJE3814-8_xenTro9",h=F,sep="\t")
scaffoldsID <- blast_BJE3814_trop9[,1]
head(scaffoldsID)
#[1] 13964 13970 13970 13970 13970 13970
scaffoldsID_unique <- unique(scaffoldsID)
head(scaffoldsID_unique)
#[1] 13964 13970 17295 18041 20125 28074
number_scaffolds_match_BJE3814_XenTrop9 <- length(scaffoldsID_unique)
number_scaffolds_match_BJE3814_XenTrop9
#[1] 38797
```
In `BJE3814-stats.tab` from Abyss output: total number of scaffolds: 7557165 -> 38797/7557165 = 0.5% (FEMALE against xenTrop9). For BJE3815 (MALE): 38928/7481189 = 0.5%
BJE3815 blast against BJE3814: 1446606/7481189 = 19%
BJE3814 blast against BJE3815: 1442231/7557165 = 19%

For e-20:
BJE3814 blast against xenTrop9: 167698/7557165 = 2%
BJE3815 blast against xenTrop9: 165767/7481189 = 2%

##Blast Sox3 of *X. tropicalis* against both *de novo* assambly
[cDNA & coding region](http://useast.ensembl.org/Xenopus_tropicalis/Gene/Summary?db=core;g=ENSXETG00000007607;r=GL172698.1:1304016-1305642;t=ENSXETT00000016558) (coding region tend to be more conserved) and total gene because good match (NCBI Reference Sequence: NW_004668241.1 - XenTrop7)

```
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_Sox3_cDNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Sox3_cDNA -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis7_Sox3_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Sox3_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_ar_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_ar_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_dmrta1_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_dmrta1_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_dmrta2_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_dmrta2_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Bufo_dmrt1_genebank.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Bufo_dmrt1 -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_Cyp17_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_xenTro_Cyp17_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/DMW_laevis_exon4.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_laevis_DMW_e4 -outfmt 6 -max_target_seqs 1
```

|Genes (if not specified: X. trop)|Sox3 | AR | Dmrt1-bufo | dmrta1 | dmrta2 | Cyp17 | Cyp19 | foxl2 | DMW-e2-laevis | DMW-e3-laevis | DMW-e4-laevis | 
|---|---|---|---|---|---|---|---|---|---|---|---|
|Female | &#10004;|&#10004;|&#88;|&#88;|&#88;|&#10004;|e-10|&#10004;|&#88;|&#88;|&#88;|
|Male |&#10004;|&#10004;|&#88;|&#88;|&#88;|&#10004;|e-10|&#10004;|&#88;|&#88;|&#88;|

(Try with `-evalue 1e-10` for dmrta1-bufo/-trop/dmrta2/Cyp19/DMW: Cyp19 in female+male). 

See [Uno et al. 2008](http://www.ncbi.nlm.nih.gov/pubmed/18484182) for more sex-determining genes:  AR, SF-1/Ad4BP, Sox3, Dmrt1 (might not related to sex determination in *Rana rugosa*). And [Nakamura 2009](http://www.ncbi.nlm.nih.gov/pubmed/18996493): CYP19, CYP17, Foxl2.

> Compare size + identity of the blast results between male/female.

## Stampy (because Cortex not working for the moment)
```
/work/ben/stampy-1.0.28/stampy.py -t8 -g /work/ben/2016_Hymenochirus/xenTro7/xenTro7 -h /work/ben/2016_Hymenochirus/xenTro7/xenTro7 --substitutionrate=0.10 -o /work/ben/2016_Hymenochirus/BJE3814/BJE3814_stampy_xenTro7_paired.sam -M /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz
```
Estimate coverage among the reference genome (samtools mpileup) + plot with `R`.

```
/work/ben/macaque_RAD_TAGs/samtools-0.1.18/samtools flagstat BJE3815_stampy_xenTro7_paired_sorted.bam > BJE3815_stampy_xenTro7_paired_sorted_flagstat.txt
```
Output from flagstat from BJE3815
```
87370213 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
5278107 + 0 mapped (6.04%:-nan%)
87370213 + 0 paired in sequencing
43685107 + 0 read1
43685106 + 0 read2
1261700 + 0 properly paired (1.44%:-nan%)
2290444 + 0 with itself and mate mapped
2987663 + 0 singletons (3.42%:-nan%)
793302 + 0 with mate mapped to a different chr
121082 + 0 with mate mapped to a different chr (mapQ>=5)
```
Output from flagstat from BJE3814
```
102080926 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
6247778 + 0 mapped (6.12%:-nan%)
102080926 + 0 paired in sequencing
51040463 + 0 read1
51040463 + 0 read2
1502320 + 0 properly paired (1.47%:-nan%)
2757848 + 0 with itself and mate mapped
3489930 + 0 singletons (3.42%:-nan%)
969546 + 0 with mate mapped to a different chr
144536 + 0 with mate mapped to a different chr (mapQ>=5)
```

> Results Stampy: bad mapping -> use bwa or bowtie to map male & female reads against Abyss genome (of both sex) and infer the coverage using Samtools > use bwa mem command "For 70bp or longer Illumina, 454, Ion Torrent and Sanger reads, assembly contigs and BAC sequences, BWA-MEM is usually the preferred algorithm."  from [bwa website](http://bio-bwa.sourceforge.net/)

```perl
#!/usr/bin/perl                                                                                                                                                   

use warnings;
use strict;

# This script will execute alignment functions for paired end reads
# using bwa mem for reference contigs

my $path_to_bwa="/work/ben/macaque_RAD_TAGs/bwa-0.6.2";
my $path_to_samtools="/work/ben/macaque_RAD_TAGs/samtools-0.1.18";
my $path_to_data="/work/ben/2016_Hymenochirus/BJE3814";
my $path_to_genome="/work/ben/2016_Hymenochirus/BJE3815";
my $genome="BJE3815-8.fa";
my $individual=" BJE3814";

my $commandline;
my $status;

# Index
$commandline = $path_to_bwa."\/bwa index -a bwtsw ".$path_to_genome."\/".$genome;

$status = system($commandline);

# bwa mem
$commandline = $path_to_bwa."\/bwa mem ".$path_to_genome. "\/".$genome. $path_to_data."\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.gz".$path_to_data."\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.sam";
$status = system($commandline);

# Samtools : sam to bam
$commandline=$path_to_samtools."\/samtools view -bt ".$path_to_genome."\/".$genome." -o ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.bam ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.sam";
$status = system($commandline);
$commandline=$path_to_samtools."\/samtools sort ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.bam ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815_sorted";
$status = system($commandline);
$commandline= $path_to_samtools."\/samtools index ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815_sorted.bam";
$status = system($commandline);
```
# SOAP de novo 
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s /work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/hymeno.config -K 64 -R -o SOAP_Hymeno_genome 1>ass.log 2>ass.err
```
## SOAP chimeric genome (reads from mom + dad) - statistics for the scaffolds
```
Size_includeN	2818397698
Size_withoutN	2699755794
Scaffold_Num	9612629
Mean_Size	293
Median_Size	130
Longest_Seq	116517
Shortest_Seq	100
Singleton_Num	9021371
Average_length_of_break(N)_in_scaffold	12

scaffolds>100 	9426455	98.06%
scaffolds>500 	769632	8.01%
scaffolds>1K  	386272	4.02%
scaffolds>10K 	16535	0.17%
scaffolds>100K	3	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	832817850	29.55%
Nucleotide_C	538677655	19.11%
Nucleotide_G	521887058	18.52%
Nucleotide_T	806373231	28.61%
GapContent_N	118641904	4.21%
Non_ACGTN	0	0.00%
GC_Content	39.28%		(G+C)/(A+C+G+T)

N10	9297	19312
N20	4885	62312
N30	2624	141850
N40	1294	297103
N50	631	615567
N60	292	1285271
N70	161	2606172
N80	130	4703979
N90	116	6991582
```
# Mapping with `bwa` 

Ref genome = `X. tropicalis v9.0 genome assembly` from [XenBase](http://www.xenbase.org/other/static/ftpDatafiles.jsp)
```perl
#!/usr/bin/perl                                                                                                             

use warnings;
use strict;

# This script will execute alignment functions for paired end reads                                                          
# using bwa mem for reference contigs                                                                                        

my $path_to_bwa="/work/ben/macaque_RAD_TAGs/bwa-0.6.2";
my $path_to_samtools="/work/ben/macaque_RAD_TAGs/samtools-0.1.18";
my $path_to_data="/work/ben/2016_Hymenochirus/BJE3814";
my $path_to_genome="/work/ben/2016_Hymenochirus/xenTro9";
my $genome="Xtropicalis_v9_repeatMasked.fa";
my $individual="BJE3814";

my $commandline;
my $status;

# Index                                                                                                                      
#$commandline = $path_to_bwa."\/bwa index -a bwtsw ".$path_to_genome."\/".$genome;                                           

#$status = system($commandline);                                                                                             

#bwa sampe                                                                                                                   
$commandline = $path_to_bwa."\/bwa aln ".$path_to_genome."\/".$genome." ".$path_to_data."\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.sai";
$status = system($commandline);
$commandline = $path_to_bwa."\/bwa aln ".$path_to_genome."\/".$genome." ".$path_to_data."\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.sai";
$status = system($commandline);

# bwa sampe                                                                                                                  
$commandline = $path_to_bwa."\/bwa sampe ".$path_to_genome. "\/".$genome." ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.sai ".$path_to_data."\/map_to_abyss_genome\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.sai ".$path_to_data."\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.gz ".$path_to_data."\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9.sam";
$status = system($commandline);

# Samtools : sam to bam                                                                                                      
$commandline=$path_to_samtools."\/samtools view -bt ".$path_to_genome."\/".$genome." -o ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_XenTro9.bam ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9.sam";
$status = system($commandline);
$commandline=$path_to_samtools."\/samtools sort ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9.bam ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9_sorted";
$status = system($commandline);
$commandline= $path_to_samtools."\/samtools index ".$path_to_data."\/map_to_abyss_genome\/".$individual."_XenTro9_sorted.bam";
$status = system($commandline);
```
Map BJE3814 & BJE3815 to xenTrop9 and then merge .bam files and use `mpileup``bcftools``vcfutils` to have a consensus sequence
If need to add RG tags:
```
perl -e 'print "@RG\\tID:BJE3814\\tSM:hs\\tLB:BJE3814\\tPL:Illumina\\n@RG\\tID:BJE3815\\tSM:hs\\tLB:BJE3815\\tPL:Illumina\\n"' > rg.txt
samtools merge -rh rg.txt merged.bam BJE3814.bam BJE3815.bam
```
`samtools merge out.bam in1.bam in2.bam in3.bam`

`samtools mpileup [-EBugp] [-C capQcoef] [-r reg] [-f in.fa] [-l list] [-Q minBaseQ] [-q minMapQ] in.bam [in2.bam [...]]`

To have a the consensus sequence:

`samtools mpileup -uf ref.fa aln.bam | bcftools call -c | vcfutils.pl vcf2fq > cns.fq`

If mapping ok with xenTrop9 as a reference genome: use it to calculate the nucleotide diversity, otherwise use the chimere (female + male) abyss assembly. 

To do when Iqaluk OK or on info
Identify repeated sequences using `Repeatmasker` on the 2 de novo assembly (female + male) ([Burger & Palmieri 2013](http://jhered.oxfordjournals.org/content/105/6/933.full) an example for using Repeatmasker   
Compare with repeat elements identified in [X. trop](http://www.repeatmasker.org/genomicDatasets/RMGenomicDatasets.html)

- Reads mapping against XenTrop9 genome
```
bwa mem /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa>aln_bwa_mem_trop9_BJE3814.sam

samtools view -bt /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa -o /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814.bam /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814.sam

samtools sort /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814.bam /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814_sorted

samtools index /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814_sorted.bam

samtools flagstat /home/caroline/hymeno/BWA_analysis/aln_bwa_mem_trop9_BJE3814_sorted.bam > aln_bwa_mem_trop9_BJE3814_sorted_flagstat.txt
```
- Reads mapping against their own abyss genome
```
bwa index /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_BJE3814_reads_abyss.sam

samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_BJE3814_reads_abyss.sam

samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss_sorted

samtools index /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss_sorted.bam

samtools flagstat /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814_reads_abyss_sorted.bam > aln_bwa_mem_BJE3814_reads_abyss_sorted_flagstat.txt
```
On info bwa: Version: 0.7.8-r455

- Reads mapping against the other abyss genome (ex.: reads female against male abyss genome...)
```
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_BJE3814abyss_BJE3815reads.sam

bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3815-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_BJE3815abyss_BJE3814reads.sam

samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_BJE3814abyss_BJE3815reads.sam

samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted

samtools index /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted.bam

samtools flagstat /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted.bam > aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted_flagstat.txt
```
- Estimate the coverage for each abyss genome (female or male) using the 2 types of read (female + male)
```
samtools depth aln_bwa_mem_BJE3814abyss_BJE3815reads_sorted.bam aln_bwa_mem_BJE3814_reads_abyss_sorted.bam > abyssBJE3814_depth_BJE3814_BJE3815.depth
```
- Reads mapping against the chimeric SOAP de novo genome
```
bwa index /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_ChimereSOAP1415_BJE3814reads.sam

samtools view -bt /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq -o /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3814reads.bam /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/samfiles/aln_bwa_mem_ChimereSOAP1415_BJE3814reads.sam

samtools sort /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3814reads.bam /home/evanslab/Hymeno_fastqc/map_reads_to_abyss_genomes/aln_bwa_mem_ChimereSOAP1415_BJE3814reads_sorted

TO RUN when others finished:
bwa mem /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz > aln_bwa_mem_ChimereSOAP1415_BJE3815reads.sam
```
## [MUMMER](http://mummer.sourceforge.net/manual/#description)

- Can be used for align multiple finished or draft genomes and identify at some point some repeats regions (align seq against itself)

Us: Reference (XenTrop9) divergent from the 2 other sequences (BJE3814 & BJE3815). If not working just nucmer on Abyss Assembly (BJE3814 & BJE3815) to identify similar and different regions. (ON INFO)
```
promer --prefix=ref_qry ref.fasta qry.fasta
/usr/local/mummer/MUMmer3.23/promer -o --prefix=xentrop9_BJE3814 Xtropicalis_v9_repeatMasked.fa BJE3814-8.fa 
/usr/local/mummer/MUMmer3.23/mummer -L -c Xtropicalis_v9_repeatMasked.fa BJE3814-8.fa BJE3815-8.fa

/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/nucmer -o --prefix=Abyss_BJE3814_BJE3815 /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa /home/caroline/hymeno/MUMMER_analysis/BJE3815-8.fa
/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/mummerplot /home/caroline/hymeno/MUMMER_analysis/BJE3814_3815_nucmer/Abyss_BJE3814_BJE3815.delta -R /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa -Q /home/caroline/hymeno/MUMMER_analysis/BJE3815-8.fa -S --png --prefix mummerplot_nucmer_BJE3814_BJE3815 --filter --layout

/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/promer -o --prefix=xentrop9_BJE3814 /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa
/home/caroline/hymeno/MUMMER_analysis/MUMmer3.23/mummerplot /home/caroline/hymeno/MUMMER_analysis/BJE3814_promer/xentrop9_BJE3814.delta -R /home/caroline/hymeno/MUMMER_analysis/Xtropicalis_v9_repeatMasked.fa -Q /home/caroline/hymeno/MUMMER_analysis/BJE3814-8.fa --png --prefix mummerplot_promer_xenTrop9_BJE3814 --filter --layout
```

To check if something is running on the background:
```
ps -elf | grep 'stampy'
```
