# De novo assembly

Using the data corrected with `trimmomatic` & `Quake` ([see Starting](https://github.com/caro46/Hymenochirus/blob/master/Starting.Rmd)) 

## Abyss

To unload one module:
```
module unload intel mkl openmpi
```
To load other modules:
```
module load gcc/4.9.2
module load openmpi/gcc-4.9.2/std/1.8.7
module load boost/gcc492-openmpi187/1.59.0
```
```
export PATH=/work/ben/abyss/bin:$PATH
abyss-pe np=8 name=BJE3814 k=64 in='BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz' 
```
Comments:

-Abyss for BJE3814 is finished: prepare the reference genome xenTro7 + BJE3814 as a reference for BJE3815. 

### Having a chimeric Abyss genome
```
abyss-pe np=8 k=64 name=BJEchimere lib='peFem peMale' peFem='/work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz' peMale='/work/ben/2016_Hymenochirus/BJE3815/BJE3815_S4_L004_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3815/BJE3815_S4_L004_R2_001_trim_paired.cor.fastq.gz'
```

##To blast abyss outputs
On iqaluk: blast version `2.2.28+`.
Command to use after abyss:
```
module load blast/2.2.28+
```
## Format of blast output
|QueryID | SubjectID | % id | alignment length | mistmatches | gap openings | q.start | q.end | s.start | s.end | e-value | bit score| 
|---|---|---|---|---|---|---|---|---|---|---|---| 


###Prepare the reference
Need to `gunzip` the reference genome before making it blastable.
```
makeblastdb -in /work/ben/2016_Hymenochirus/xenTro7/xenTro7.fa.masked.gz -dbtype nucl -title UCSC_xenTro7 -out /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable
```
BJE3814 for BJE3815
```
makeblastdb -in /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -dbtype nucl -title Hymeno_Female_BJE3814 -out /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable
```
###Blast the output of abyss (contigs) on *X. tropicalis* genome to identify putative sex-determining region ([female reference genome for *X. tropicalis*](http://www.ncbi.nlm.nih.gov/biosample/SAMN00000117)) 
```
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/BJE3814-8.fa -db /work/ben/2016_Hymenochirus/xenTro7/xenTro7_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTro7 -outfmt 6 -max_target_seqs 1
```
Check if contigs of our female (BJE3814) map to a region where our male (BJE3815) doesn't map.
Blast male de novo assambly against female and the opposite to find contigs only male or only female. Blast them against the female *X. tropicalis*.

On xenTrop9 (chromosomes):
```
makeblastdb -in /work/ben/2016_Hymenochirus/xenTro9/Xtropicalis_v9_repeatMasked.fa -dbtype nucl -title xenBase_xenTro9 -out /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3815/BJE3815-8.fa -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_xenTro9 -outfmt 6 -max_target_seqs 1
```
##Blast Sox3 of *X. tropicalis* against both *de novo* assambly
[cDNA & coding region](http://useast.ensembl.org/Xenopus_tropicalis/Gene/Summary?db=core;g=ENSXETG00000007607;r=GL172698.1:1304016-1305642;t=ENSXETT00000016558) (coding region tend to be more conserved) and total gene because good match (NCBI Reference Sequence: NW_004668241.1 - XenTrop7)

```
blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_Sox3_cDNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Sox3_cDNA -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis7_Sox3_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Sox3_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_ar_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_ar_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_dmrta1_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_dmrta1_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_dmrta2_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_xenTrop_dmrta2_gene -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Bufo_dmrt1_genebank.fa -db /work/ben/2016_Hymenochirus/BJE3814/BJE3814_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/BJE3814-8_Bufo_dmrt1 -outfmt 6 -max_target_seqs 1

blastn -evalue 1e-60 -query /work/ben/2016_Hymenochirus/BJE3814/blast/Xtropicalis_Cyp17_gene_DNA.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/BJE3815/blast/BJE3815-8_xenTro_Cyp17_gene -outfmt 6 -max_target_seqs 1
```

See [Uno et al. 2008](http://www.ncbi.nlm.nih.gov/pubmed/18484182) for more sex-determining genes:  AR, SF-1/Ad4BP, Sox3, Dmrt1 (might not related to sex determination in *Rana rugosa*). And [Nakamura 2009](http://www.ncbi.nlm.nih.gov/pubmed/18996493): CYP19, CYP17, Foxl2.

> Compare size + identity of the blast results between male/female.

## Stampy (because Cortex not working for the moment)
```
/work/ben/stampy-1.0.28/stampy.py -t8 -g /work/ben/2016_Hymenochirus/xenTro7/xenTro7 -h /work/ben/2016_Hymenochirus/xenTro7/xenTro7 --substitutionrate=0.10 -o /work/ben/2016_Hymenochirus/BJE3814/BJE3814_stampy_xenTro7_paired.sam -M /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /work/ben/2016_Hymenochirus/BJE3814/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz
```
Estimate coverage among the reference genome (samtools mpileup) + plot with `R`.

```
/work/ben/macaque_RAD_TAGs/samtools-0.1.18/samtools flagstat BJE3815_stampy_xenTro7_paired_sorted.bam > BJE3815_stampy_xenTro7_paired_sorted_flagstat.txt
```
Output from flagstat from BJE3815
```
87370213 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
5278107 + 0 mapped (6.04%:-nan%)
87370213 + 0 paired in sequencing
43685107 + 0 read1
43685106 + 0 read2
1261700 + 0 properly paired (1.44%:-nan%)
2290444 + 0 with itself and mate mapped
2987663 + 0 singletons (3.42%:-nan%)
793302 + 0 with mate mapped to a different chr
121082 + 0 with mate mapped to a different chr (mapQ>=5)
```
Output from flagstat from BJE3814
```
102080926 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
6247778 + 0 mapped (6.12%:-nan%)
102080926 + 0 paired in sequencing
51040463 + 0 read1
51040463 + 0 read2
1502320 + 0 properly paired (1.47%:-nan%)
2757848 + 0 with itself and mate mapped
3489930 + 0 singletons (3.42%:-nan%)
969546 + 0 with mate mapped to a different chr
144536 + 0 with mate mapped to a different chr (mapQ>=5)
```

> Results Stampy: bad mapping -> use bwa or bowtie to map male & female reads against Abyss genome (of both sex) and infer the coverage using Samtools > use bwa mem command "For 70bp or longer Illumina, 454, Ion Torrent and Sanger reads, assembly contigs and BAC sequences, BWA-MEM is usually the preferred algorithm."  from [bwa website](http://bio-bwa.sourceforge.net/)

```perl
#!/usr/bin/perl                                                                                                                                                   

use warnings;
use strict;

# This script will execute alignment functions for paired end reads
# using bwa mem for reference contigs

my $path_to_bwa="/work/ben/macaque_RAD_TAGs/bwa-0.6.2";
my $path_to_samtools="/work/ben/macaque_RAD_TAGs/samtools-0.1.18";
my $path_to_data="/work/ben/2016_Hymenochirus/BJE3814";
my $path_to_genome="/work/ben/2016_Hymenochirus/BJE3815";
my $genome="BJE3815-8.fa";
my $individual=" BJE3814";

my $commandline;
my $status;

# Index
$commandline = $path_to_bwa."\/bwa index -a bwtsw ".$path_to_genome."\/".$genome;

$status = system($commandline);

# bwa mem
$commandline = $path_to_bwa."\/bwa mem ".$path_to_genome. "\/".$genome. $path_to_data."\/".$individual."_S3_L003_R1_001_trim_paired.cor.fastq.gz".$path_to_data."\/".$individual."_S3_L003_R2_001_trim_paired.cor.fastq.gz \> ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.sam";
$status = system($commandline);

# Samtools : sam to bam
$commandline=$path_to_samtools."\/samtools view -bt ".$path_to_genome."\/".$genome." -o ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.bam ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.sam";
$status = system($commandline);
$commandline=$path_to_samtools."\/samtools sort ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815.bam ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815_sorted";
$status = system($commandline);
$commandline= $path_to_samtools."\/samtools index ".$path_to_data."\/"."map_to_abyss_genome"."\/".$individual."_genomeAbyssBJE3815_sorted.bam";
$status = system($commandline);
```

# Mapping with `bwa` 

Ref genome = `X. tropicalis v9.0 genome assembly` from [XenBase](http://www.xenbase.org/other/static/ftpDatafiles.jsp)

Map BJE3814 & BJE3815 to xenTrop9 and then merge them to have a consensus sequence 
`samtools merge out.bam in1.bam in2.bam in3.bam`
`samtools mpileup [-EBugp] [-C capQcoef] [-r reg] [-f in.fa] [-l list] [-Q minBaseQ] [-q minMapQ] in.bam [in2.bam [...]]`

If mapping ok with xenTrop9 as a reference genome: use it to calculate the nucleotide diversity, otherwise use the chimere (female + male) abyss assembly. 

To do when Iqaluk OK or on info
Identify repeated sequences using `Repeatmasker` on the 2 de novo assembly (female + male) ([Burger & Palmieri 2013](http://jhered.oxfordjournals.org/content/105/6/933.full) an example for using Repeatmasker   


## [MUMMER](http://mummer.sourceforge.net/manual/#description)

- Can be used for align multiple finished or draft genomes and identify at some point some repeats regions (align seq against itself)

Us: Reference (XenTrop9) divergent from the 2 other sequences (BJE3814 & BJE3815). If not working just nucmer on Abyss Assembly (BJE3814 & BJE3815) to identify similar and different regions.
```
promer --prefix=ref_qry ref.fasta qry.fasta
promer -o --prefix=xentrop9_BJE3814_3815 Xtropicalis_v9_repeatMasked.fa BJE3814-8.fa BJE3815-8.fa
```

To check if something is running on the background:
```
ps -elf | grep 'stampy'
```
