#Amplification of putative sex-linked genes

See [there](https://github.com/caro46/Hymenochirus/blob/master/Optimization_SL_identification.Rmd) to see how to find the putative genes. 
Now the goal is to amplify by Sanger sequencing the regions with a sites which displays a sex-inheritance pattern in order to confirm this pattern.
I am using primer3 included on [NCBI](https://www.ncbi.nlm.nih.gov/tools/primer-blast/) (so can double check the primers using *X. tropicalis* to see if ok for it too)

- To amplify sall1 (need to include position 1133 of the scaffold97784: forward primer from 500, reverse primer to 1500): 
```
awk -v seq="scaffold97784" -v RS='>' '$1 == seq {print RS $0}' /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/Hymenochirus_putative_sex_linked_opt_tot1SS1polym1ratio0_repARKmasked.fa.masked >scaffold97784.fa
```
```
Primer pair 3
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	AAATGAAACCCGCCCCATCC	Plus	20	999	1018	60.98	55.00	3.00	0.00
Reverse primer	CCAGCAGACAACCGGAGAAA	Minus	20	1269	1250	60.25	55.00	4.00	0.00
Product length	271

Primer pair 5
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	ACTAAATGAAACCCGCCCCAT	Plus	21	996	1016	59.99	47.62	3.00	2.00
Reverse primer	CCGGAGAAACACGGACAACA	Minus	20	1258	1239	60.53	55.00	4.00	0.00
Product length	263

Primer pair 6
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CCCCATCCCCATAACAAACGA	Plus	21	1011	1031	60.06	52.38	2.00	0.00
Reverse primer	CAACCGGAGAAACACGGACA	Minus	20	1261	1242	60.53	55.00	4.00	0.00
Product length	251
```
If look for [specificity](https://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1485445334&job_key=CwHUHTvoNkARfix7IRsISVsAGXt2EwJmdw&CheckStatus=Check) of primers from *Xenopus* species:
```
Primer pair 4

	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	ACTAAATGAAACCCGCCCCAT	Plus	21	996	1016	59.99	47.62	3.00	2.00
Reverse primer	CCGGAGAAACACGGACAACA	Minus	20	1258	1239	60.53	55.00	4.00	0.00
Product length	263

Primer pair 5
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CCCCATCCCCATAACAAACGA	Plus	21	1011	1031	60.06	52.38	2.00	0.00
Reverse primer	CAACCGGAGAAACACGGACA	Minus	20	1261	1242	60.53	55.00	4.00	0.00
Product length	251

Primer pair 6
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GCCCCATCCCCATAACAAAC	Plus	20	1010	1029	58.88	55.00	2.00	0.00
Reverse primer	ACAACCGGAGAAACACGGAC	Minus	20	1262	1243	60.53	55.00	4.00	1.00
Product length	253

Primer pair 8
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GAACTAAATGAAACCCGCCCC	Plus	21	994	1014	59.52	52.38	2.00	0.00
Reverse primer	CGGAGAAACACGGACAACAA	Minus	20	1257	1238	58.71	50.00	2.00	0.00
Product length	264

Primer pair 10
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TAAATGAAACCCGCCCCATCC	Plus	21	998	1018	60.69	52.38	3.00	0.00
Reverse primer	AACCAGCAGACAACCGGAGA	Minus	20	1271	1252	61.41	55.00	4.00	0.00
Product length	274
```
Primer pair 8 seems the best.

- To amplify ptprf (need to include position 5250 of the scaffold: forward primer from , reverse primer to )

*! position masked by the repeat*

Note: Need to discuss that with Ben boss, since the library was made based on k-mer distribution, it makes some sense that some regions near restriction sites might be considered as repetitive and masked...
```
awk -v seq="scaffold147062" -v RS='>' '$1 == seq {print RS $0}' /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/Hymenochirus_putative_sex_linked_opt_tot1SS1polym1ratio0_repARKmasked.fa.masked >SOAP_chimerical_scaffold147062_masked.fa
```
- To amplify [atrx](https://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1485356454&job_key=wsgdshEUHLw7ghmHFOc9tW78LIdD7zeaQg&CheckStatus=Check) (need to include position 15301 of the scaffold75704: forward primer from 15100, reverse primer to 16000)

```
awk -v seq="scaffold75704" -v RS='>' '$1 == seq {print RS $0}' /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/Hymenochirus_putative_sex_linked_opt_tot1SS1polym1ratio0_repARKmasked.fa.masked >SOAP_chimerical_scaffold75704_masked.fa 
```
```
Primer pair 2
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TGACCCAGGACTTCCATCCA	Plus	20	15173	15192	60.18	55.00	4.00	0.00
Reverse primer	AATTCCAGGCTCTTGAGCGG	Minus	20	15950	15931	60.39	55.00	6.00	2.00
Product length	778

Primer pair 3
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	AGGACTTCCATCCACGTTCG	Plus	20	15179	15198	59.75	55.00	4.00	2.00
Reverse primer	ACTGCAGATGTCGGGACAAA	Minus	20	15547	15528	59.60	50.00	6.00	2.00
Product length	369

Primer pair 4
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CTTGCTTGAAAGCTTGGTCCC	Plus	21	15148	15168	60.00	52.38	6.00	2.00
Reverse primer	CAGGCTCTTGAGCGGTTATCT	Minus	21	15945	15925	59.86	52.38	6.00	0.00
Product length	798

Primer pair 5
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GCTTGGTCCCTTTATGACCCA	Plus	21	15159	15179	59.99	52.38	4.00	2.00
Reverse primer	TGTTTTGCAGTCCAGGTGGTA	Minus	21	15765	15745	59.79	47.62	4.00	2.00
Product length	607

Primer pair 6
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TTGCTTGAAAGCTTGGTCCC	Plus	20	15149	15168	58.96	50.00	6.00	2.00
Reverse primer	TGGGTCTGTTGACCTTTCCA	Minus	20	15675	15656	58.78	50.00	4.00	0.00
Product length	527

Primer pair 7
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TGGCACTGTGAGATCAACCT	Plus	20	15239	15258	58.94	50.00	5.00	2.00
Reverse primer	ATTCCAGGCTCTTGAGCGG	Minus	19	15949	15931	59.78	57.89	6.00	2.00
Product length	711

Primer pair 8
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GCAACTTGCAGACTTGCTTGA	Plus	21	15136	15156	59.93	47.62	8.00	1.00
Reverse primer	AGGCTCTTGAGCGGTTATCTC	Minus	21	15944	15924	59.59	52.38	6.00	1.00
Product length	809
```
If check [specificity](https://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1485445749&job_key=4es-luZR6_nMw3vGdqZf9Ay9TsYhrlXbIA) in *Xenopus*:
```
Primer pair 4
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TGGTCCCTTTATGACCCAGGA	Plus	21	15162	15182	60.20	52.38	4.00	1.00
Reverse primer	CGAACAGAATAGAGCCCCAGT	Minus	21	15336	15316	59.52	52.38	2.00	2.00
Product length	175

Primer pair 5
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GGTCCCTTTATGACCCAGGAC	Plus	21	15163	15183	59.79	57.14	4.00	2.00
Reverse primer	ACGAACAGAATAGAGCCCCAG	Minus	21	15337	15317	59.52	52.38	2.00	1.00
Product length	175

Primer pair 6
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	ATGACCCAGGACTTCCATCC	Plus	20	15172	15191	58.78	55.00	4.00	1.00
Reverse primer	TGGGTCTGTTGACCTTTCCAA	Minus	21	15675	15655	59.43	47.62	4.00	2.00
Product length	504

Primer pair 7
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CCCAGGACTTCCATCCACG	Plus	19	15176	15194	59.78	63.16	4.00	2.00
Reverse primer	GGTCTGTTGACCTTTCCAAGC	Minus	21	15673	15653	59.39	52.38	4.00	2.00
Product length	498

Primer pair 8
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GCTTGAAAGCTTGGTCCCTT	Plus	20	15151	15170	58.67	50.00	6.00	0.00
Reverse primer	CCAGGCTCTTGAGCGGTTATC	Minus	21	15946	15926	60.81	57.14	6.00	2.00
Product length	796

Primer pair 9
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TGCTTGAAAGCTTGGTCCCTT	Plus	21	15150	15170	60.41	47.62	6.00	0.00
Reverse primer	TTCCAGGCTCTTGAGCGGTT	Minus	20	15948	15929	61.77	55.00	6.00	1.00
Product length	799

```
Best: 4, 5, 6, 7 considering complementarity. But 9 seems ok.

- To amplify ifitm1 (need to include position 2439 of the scaffold428256: forward primer from , reverse primer to )

*! position masked by the repeat*

```
awk -v seq="scaffold428256" -v RS='>' '$1 == seq {print RS $0}' /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/Hymenochirus_putative_sex_linked_opt_tot1SS1polym1ratio0_repARKmasked.fa.masked >SOAP_chimerical_scaffold428256_masked.fa
```
```
```
- To amplify ppp6r3 (need to include position 184 of the scaffold34529: forward primer from , reverse primer to )

*! position masked by the repeat*

- To amplify [hmcn1](https://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1485376424&job_key=ZW-6eSOWLj4JBDgBNWEcM096DQFiaRYcYw) (need to include position 8102 of the scaffold40934: forward primer from 7000, reverse primer to 8500)
```
awk -v seq="scaffold40934" -v RS='>' '$1 == seq {print RS $0}' /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/Hymenochirus_putative_sex_linked_opt_tot1SS1polym1ratio0_repARKmasked.fa.masked >scaffold40934.fa
```
```
Primer pair 2
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CTGCAGGTTACACCTTTCGG	Plus	20	8093	8112	58.84	55.00	6.00	2.00
Reverse primer	TTCTTTGCTGCCTCCCTTCAT	Minus	21	8199	8179	59.92	47.62	3.00	2.00
Product length	107

Primer pair 5
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CCTGCAGGTTACACCTTTCG	Plus	20	8092	8111	58.84	55.00	8.00	2.00
Reverse primer	TGCTGCCTCCCTTCATGTTTA	Minus	21	8194	8174	59.65	47.62	4.00	2.00
Product length	103

Primer pair 6
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TAGCAACTAATGCTGCTGGGG	Plus	21	7838	7858	60.41	52.38	5.00	0.00
Reverse primer	GACCGAAAGGTGTAACCTGC	Minus	20	8114	8095	58.85	55.00	6.00	2.00
Product length	277

Primer pair 7
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TCCTCTCCTGCAGGTTACAC	Plus	20	8086	8105	58.73	55.00	8.00	2.00
Reverse primer	TTTGCTGCCTCCCTTCATGTT	Minus	21	8196	8176	60.48	47.62	4.00	0.00
Product length	111

Primer pair 9
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CAACTAATGCTGCTGGGGAAG	Plus	21	7841	7861	59.25	52.38	3.00	0.00
Reverse primer	TGACCGAAAGGTGTAACCTGC	Minus	21	8115	8095	60.54	52.38	6.00	2.00
Product length	275
Primer pair 10
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GCAGGTTACACCTTTCGGTCA	Plus	21	8095	8115	60.54	52.38	6.00	1.00
Reverse primer	TGCTGCCTCCCTTCATGTT	Minus	19	8194	8176	59.23	52.63	4.00	0.00
Product length	100
```
Checking *Xenopus* target specificity
```
Primer pair 5
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CCTGCAGGTTACACCTTTCG	Plus	20	8092	8111	58.84	55.00	8.00	2.00
Reverse primer	TGCTGCCTCCCTTCATGTTTA	Minus	21	8194	8174	59.65	47.62	4.00	2.00
Product length	103

Primer pair 6
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TAGCAACTAATGCTGCTGGGG	Plus	21	7838	7858	60.41	52.38	5.00	0.00
Reverse primer	GACCGAAAGGTGTAACCTGC	Minus	20	8114	8095	58.85	55.00	6.00	2.00
Product length	277

Primer pair 8
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TCCTGCAGGTTACACCTTTCG	Plus	21	8091	8111	60.27	52.38	8.00	2.00
Reverse primer	TTCTGTTCTTTGCTGCCTCC	Minus	20	8204	8185	58.39	50.00	3.00	0.00
Product length	114

Primer pair 11
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	CAACTAATGCTGCTGGGGAAG	Plus	21	7841	7861	59.25	52.38	3.00	0.00
Reverse primer	TGACCGAAAGGTGTAACCTGC	Minus	21	8115	8095	60.54	52.38	6.00	2.00
Product length	275

```
Best: 11.

##### Genes we don't know the location on *X. tropicalis* and *X. laevis*
- To amplify vps9d1 (need to include position 571 of the scaffold C92080241: forward primer from , reverse primer to )

```
awk -v seq="C92080241" -v RS='>' '$1 == seq {print RS $0}' /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/Hymenochirus_putative_sex_linked_opt_tot1SS1polym1ratio0_repARKmasked.fa.masked >SOAP_chimerical_C92080241_masked.fa
```
Scaffold small, so directly done with *Xenopus* [specificity](https://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1485447018&job_key=dnypYEfpSkFtf1B6XRp0SCcBZXoKEn5nCw)
```
Primer pair 7
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GCTTGGCATACAATCGGTTTGA	Plus	22	112	133	59.84	45.45	5.00	3.00
Reverse primer	TGGCAAGTGATAAAGGGAGGG	Minus	21	635	615	59.72	52.38	3.00	0.00
Product length	524

Primer pair 8
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TCGACTGACTGCCCAGAAATA	Plus	21	160	180	58.82	47.62	4.00	2.00
Reverse primer	GGGTTCTCAGGTCAGTGTCT	Minus	20	614	595	58.66	55.00	3.00	1.00
Product length	455

Primer pair 9
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TATTCGACTGACTGCCCAGAA	Plus	21	157	177	58.82	47.62	4.00	0.00
Reverse primer	AACCCCAGACAGGCTTCACA	Minus	20	666	647	61.35	55.00	3.00	0.00
Product length	510
```
Best:8.

##### Genes we know the locations (chr.4) but genotypes not next to them
And also do DMRTa2 (but need to produce the primers using *X. tropicalis* because small scaffold for *Hymenochirus*)

Using NBCI refseq [NM_001103073.1](https://www.ncbi.nlm.nih.gov/gene/100125187#reference-sequences) [see also](https://www.ncbi.nlm.nih.gov/protein/NP_001096543.1). [CDS](https://www.ncbi.nlm.nih.gov/nuccore/156718001) starting at position 604, the DM was identified from Location:45 → 91 (of protein from *X. tropicalis* = 46aa -> 138bp), forward primer from 500, reverse primer to 1000, primer should span an exon-exon junction, Primer Pair Specificity Checking Parameters (organism: Pipidae). [Primers](https://www.ncbi.nlm.nih.gov/tools/primer-blast/primertool.cgi?ctg_time=1485373779&job_key=rKZzsOqW5z7ABPEB_GHVM4Z6xAGrad8cqg)
```
Primer pair 1
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GTGAGTGGGTAAGGCTGGG	Plus	19	500	518	59.70	63.16	3.00	0.00
Reverse primer	CGCGATCAGCGTACACTTTG	Minus	20	849	830	59.98	55.00	6.00	2.00
Product length	350
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 2
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TGAGTGGGTAAGGCTGGGG	Plus	19	501	519	60.92	63.16	3.00	0.00
Reverse primer	GGGATACTTCTCTGCGGCTC	Minus	20	735	716	59.97	60.00	3.00	1.00
Product length	235
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 3
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GTGAGTGGGTAAGGCTGGGG	Plus	20	500	519	62.20	65.00	3.00	0.00
Reverse primer	CTGGGGGTCCATTCAGTTCC	Minus	20	625	606	60.03	60.00	3.00	0.00
Product length	126
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 4
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GAGTGGGTAAGGCTGGGG	Plus	18	502	519	59.00	66.67	3.00	0.00
Reverse primer	CAGTCTTTCCAGCGGCAGTA	Minus	20	821	802	60.04	55.00	3.00	3.00
Product length	320
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 5
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TGGGTAAGGCTGGGGCA	Plus	17	505	521	60.18	64.71	3.00	1.00
Reverse primer	TACAGTCTTTCCAGCGGCAG	Minus	20	823	804	60.04	55.00	3.00	3.00
Product length	319
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 6
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GGGTAAGGCTGGGGCAC	Plus	17	506	522	59.67	70.59	3.00	2.00
Reverse primer	TTGGGGGTACGGGGATACTT	Minus	20	746	727	59.96	55.00	4.00	0.00
Product length	241
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 7
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GTGGGTAAGGCTGGGGC	Plus	17	504	520	59.67	70.59	3.00	2.00
Reverse primer	GGGGGTACGGGGATACTTCT	Minus	20	744	725	60.11	60.00	4.00	0.00
Product length	241
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 8
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	AGTGGGTAAGGCTGGGGC	Plus	18	503	520	61.34	66.67	3.00	2.00
Reverse primer	GCCACTGTGACTGGAAGTGA	Minus	20	680	661	59.89	55.00	5.00	3.00
Product length	178
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 9
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	GAGTGGGTAAGGCTGGGGC	Plus	19	502	520	62.35	68.42	3.00	2.00
Reverse primer	CGGTAGAAGTTGCGGTCTGA	Minus	20	658	639	59.76	55.00	3.00	3.00
Product length	157
Exon junction	512/513 (forward primer) on template NM_001103073.1

Primer pair 10
	Sequence (5'->3')	Template strand	Length	Start	Stop	Tm	GC%	Self complementarity	Self 3' complementarity
Forward primer	TGGGTAAGGCTGGGGCAC	Plus	18	505	522	61.66	66.67	3.00	2.00
Reverse primer	GCGATCAGCGTACACTTTGC	Minus	20	848	829	60.25	55.00	6.00	2.00
Product length	344
Exon junction	512/513 (forward primer) on template NM_001103073.1
```
Best: seems 5.


###Find genes involved in sex-determination within our region determined as potentially sex-linked
Chr. 4: limit petit/long arms: znf423 (Chr.4: 46965827..47078060 - v.9). 
Considering results from `Hymenochirus_chimerical_putative_sex_linked_Phred20_Haplotype_caller_linked_opt_tot1SS1polym1ratio0_RepARKmasked_xenTro9_hard_mask_e5_no_max_target` (same used for identifying the genes): 4 scaffolds on XTR4p, 7 on XTR4q. DMRTa2 is on the long arm (85796936..85801496). SOX6 on small arm.

Need to see if with HiSeq genotype calling we observe a difference between male and female for DMRTa2. OK, already did that previously and for the scaffold that have DMRTa2 on: 61 or 67 sites different between mother and father, everytime dad heterozygous and mother homozygous (considering male and female mapping against female abyss and male and female mapping against male abyss).  
```
vcftools --vcf ~/Hymeno_fastqc/GATK/BJE3815_nonrecal_varonly.vcf --chr "supercontig_6" --from-bp 22045521 --to-bp 22062501 --recode --recode-INFO-all --out /4/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex/bwa_results/Phred20/all/GVCF/multiple_steps/BJE3815_nonrecal_varonly_scaffold24873906
```
####Where are the SNPs

On abyss genome BJE3814: DM-domain blasted 8211-8149 (=supercontig_5:182770532-182770470). DM domain in exon 2. Using results from blastn on Ensembl: Exon3 (182762321+6045=182768366 ... 182762321+6914 = 182769235): exon 2 (182762321+7940=182770261 ... 182762321+8190=182770511):, exon1: x
Closest SNPs form DM domain: 
```
supercontig_5   182769745       C       C/C     C/T
supercontig_5   182771619       G       G/GTA   GTA/GTATATA
```
Intron 2-3 
```
supercontig_5   182768044       C       C/C     C/T
supercontig_5   182768156       A       A/A     A/G
supercontig_5   182769411       C       C/C     C/T
supercontig_5   182769745       C       C/C     C/T
```
Upstream region: ~155bp downstream
```
supercontig_5   182768156       A       A/A     A/G
```
```
blastn -evalue 1e-1 -query /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/exon1_DMRTa3_xenbase_v8.fa -db /work/ben/2016_Hymenochirus/BJE3815/BJE3815_genomeAbyss_blastable -out /work/ben/2016_Hymenochirus/putative_sex_linked_regions_blast_Haplotype_caller/BJE3815_abyss_DMRTA2_exon1_e1_target_seq_3 -outfmt 6 -max_target_seqs 3
```
Cannot get exon 1.

Should amplify intron 2-3 to get SNPs identified and also the DM domain.
