# Genotype calls: GATK

[O'Rawe et al. 2013](http://www.ncbi.nlm.nih.gov/pubmed/23537139) reviewed some variant calling pipelines. They highligted a poor concordance of SNP and INDEL calls between pipelines, but also that a part of one-pipeline-only calling can be validated.
We will use [GATK](https://www.broadinstitute.org/gatk/) & [samtools mpileup](http://samtools.sourceforge.net/mpileup.shtml) for variant callings. GATK tries to improve genotype calling by realigning indels, recalibrating base quality... 

We will use our chimeric genome (female + male) produced by SOAP de novo as a reference genome (*X. tropicalis* too divergent), if its quality is OK (check by mapping the reads).
## Reference - preparation
```
samtools faidx /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq.fa
```
```
java -jar /usr/local/picard-tools-1.131/picard.jar CreateSequenceDictionary REFERENCE=/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq.fa OUTPUT=/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.dict
```
(picard needs .fa or .fa.gz as an extension: `SOAP_Hymeno_genome.scafSeq.fa` just obtained by `scp SOAP_Hymeno_genome.scafSeq SOAP_Hymeno_genome.scafSeq.fa`). Error message `Exception in thread "main" java.lang.OutOfMemoryError: Requested array size exceeds VM limit` because of memory, try to use `-Xmx20G` to increase memory for hash, still getting the error message. Probably due to the high number of scaffolds in Abyss genome: need to use a `perl` script to build big scaffolds. Or/and see how `samtools mpileup` deals with that.
```
java -Xmx20G -jar /usr/local/picard-tools-1.131/picard.jar CreateSequenceDictionary REFERENCE=/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq.fa OUTPUT=/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/SOAP_Hymeno_genome.scafSeq.dict
```
We made bigger scaffolds for Abyss genomes and try again (no pb anymore for making the `.dict` file
```
samtools faidx /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8_abyss_supercontigs.fa
java -jar /usr/local/picard-tools-1.131/picard.jar CreateSequenceDictionary REFERENCE=/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8_abyss_supercontigs.fa OUTPUT=/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8_abyss_supercontigs.dict
```
We follow the [best practices](https://www.broadinstitute.org/gatk/guide/best-practices) suggested on GATK website:

    1. PRE-PROCESSING ("alignment to a reference genome as well as some data cleanup operations")
    2. VARIANT DISCOVERY
    3. CALLSET REFINEMENT

## INDELs realignment
```perl
#!/usr/bin/perl
use warnings;
use strict;

# This script will read in the *_sorted.bam file names in a directory, and 
# make and execute a GATK commandline on these files.  

my $path_to_reference_genome="/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/";
my $reference_genome="SOAP_Hymeno_genome.scafSeq";
my @files;
my $status;

@files = glob("aln_bwa_mem_ChimereSOAP1415*_sorted.bam");

my $commandline = "java -Xmx1G -jar /usr/local/gatk/GenomeAnalysisTK.jar -T RealignerTargetCreator ";

foreach(@files){
    $commandline = $commandline." -I ".$_." ";
}

$commandline = $commandline."-R ".$path_to_reference_genome.$reference_genome." -o forIndelRealigner.intervals";

$status = system($commandline);

my $commandline = "java -Xmx1G -jar /usr/local/gatk/GenomeAnalysisTK.jar -T IndelRealigner ";

foreach(@files){
    $commandline = $commandline." -I ".$_;
}

$commandline = $commandline." -R ".$path_to_reference_genome.$reference_genome." --targetIntervals forIndelRealigner.intervals --nWayOut _realigned.bam";

$status = system($commandline);
```
## Base recalibration

We use [Haplotype Caller](https://www.broadinstitute.org/gatk/guide/tagged?tag=haplotypecaller) to call the SNPs as suggested on GATK website.   
```
#!/usr/bin/perl
use warnings;
use strict;

# This script will read in the *_sorted_realigned.bam files, and 
# make and execute several GATK commandlines with these files that will
# (1) Use HaplotypeCaller to output a vcf file with variable sites
# (2) Use BaseRecalibrator to output a base recalibration table based on these sites
# (3) Use PrintReads to output a new concatenated bam file with Recalibrated quality scores
# (4) Use HaplotypeCaller to recall the genotypes using the new quality scores; output a new vcf file.

my $path_to_reference_genome="/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/";
my $reference_genome="BJE3814-8_abyss_supercontigs.fa";
my $path_to_data;
my $status;
my @files;

# read in the files within this directory a specific ending
$path_to_data = "/home/evanslab/Hymeno_fastqc/GATK/";
@files = glob($path_to_data."BJE3814_genomeAbyss_supercontigs_reads*sorted_rg_realigned.bam");

# construct a commandline for the HaplotypeCaller; output a vcf file with only variable sites
my $commandline = "java -Xmx1G -jar /usr/local/gatk/GenomeAnalysisTK.jar -T HaplotypeCaller -R ".$path_to_reference_genome.$reference_genome;
foreach(@files){
    $commandline = $commandline." -I ".$_." ";
}
$commandline = $commandline."-out_mode EMIT_VARIANTS_ONLY -o ./nonrecal_varonly.vcf";

# Execute this command line to call genotypes
$status = system($commandline);

# Now make a new commandline for the BaseRecalibrator, output a table for base recalibration 
$commandline = "java -Xmx3G -jar /usr/local/gatk/GenomeAnalysisTK.jar -T BaseRecalibrator -R ".$path_to_reference_genome.$reference_genome;
foreach(@files){
    $commandline = $commandline." -I ".$_." ";
}

$commandline = $commandline."-knownSites ./nonrecal_varonly.vcf -o recal_data.table";

# Execute this command line
$status = system($commandline);


# Now use PrintReads to output a new concatenated bam file with recalibrated quality scores
$commandline = "java -Xmx3G -jar /usr/local/gatk/GenomeAnalysisTK.jar -T PrintReads -R ".$path_to_reference_genome.$reference_genome;

foreach(@files){
    $commandline = $commandline." -I ".$_." ";
}

$commandline = $commandline."-BQSR recal_data.table -o concatenated_and_recalibrated_round1.bam";


$status = system($commandline);


# Now use HaplotypeCaller to recall bases with recalibrated quality scores; output a vcf file 
$commandline = "java -Xmx3G -jar /usr/local/gatk/GenomeAnalysisTK.jar -T HaplotypeCaller -R ".$path_to_reference_genome.$reference_genome;
$commandline = $commandline." -I concatenated_and_recalibrated_round1.bam";
$commandline = $commandline." -o recalibrated_round1.vcf";


$status = system($commandline);
```
