#Onemap

##Install
Add the directory where I have R packages not available on the server.
```
setenv R_LIBS $HOME/Hymeno_fastqc/R_packages
```
```
unzip onemap-master.zip #download the developmental version from github since new interesting functions like vcf2raw 
mv onemap-master onemap
tar czf onemap.tar.gz onemap

R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages onemap.tar.gz
ERROR: dependencies ‘tkrplot’, ‘fields’, ‘Rhtslib’ are not available for package ‘onemap’
```
Need to install some dependencies not available on the server.
```
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages tkrplot_0.0-23.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages fields_8.10.tar.gz 
ERROR: dependency ‘spam’ is not available for package ‘fields’
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages spam_1.4-0.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages fields_8.10.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages Rhtslib_1.0.0.tar.gz #because of autoconf version, can't install a more recent version
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages onemap.tar.gz
```
Some warning messages but seems ok when load the library into R using `library(onemap)`.
```
* installing *source* package ‘onemap’ ...
** libs
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c f2_est.cpp -o f2_est.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c find_bins.cpp -o find_bins.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c hmm_bc.cpp -o hmm_bc.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c hmm_f2.cpp -o hmm_f2.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c hmm_out.cpp -o hmm_out.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c out_est.cpp -o out_est.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c twopts_bc.cpp -o twopts_bc.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c twopts_f2.cpp -o twopts_f2.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c twopts_out.cpp -o twopts_out.o
g++ -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c utils.cpp -o utils.o
gcc -std=gnu99 -I/usr/local/R/3.2.2/lib64/R/include -DNDEBUG  -I/usr/local/include -I"/usr/local/R/3.2.2/lib64/R/library/Rcpp/include" -I"/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/include" -I"/usr/local/R/3.2.2/lib64/R/library/zlibbioc/include"   -fpic  -g -O2  -c vcf2raw.c -o vcf2raw.o
g++ -shared -L/usr/local/R/3.2.2/lib64/R/lib -L/usr/local/lib64 -o onemap.so f2_est.o find_bins.o hmm_bc.o hmm_f2.o hmm_out.o out_est.o twopts_bc.o twopts_f2.o twopts_out.o utils.o vcf2raw.o -L/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/lib -Wl,-rpath,/home/evanslab/Hymeno_fastqc/R_packages/Rhtslib/lib -lhts -lz -pthread -L/usr/local/R/3.2.2/lib64/R/lib -lR
installing to /home/evanslab/Hymeno_fastqc/R_packages/onemap/libs
** R
** data
** inst
** preparing package for lazy loading
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection refused
Warning in fun(libname, pkgname) :
  couldn't connect to display "localhost:10.0"
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection timed out
Warning: loading Rplot failed
** help
*** installing help indices
** building package indices
** installing vignettes
   ‘Inbred_Based_Populations.Rmd’ using ‘UTF-8’ 
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection refused
Warning in grDevices::png(tempfile()) :
  unable to open connection to X11 display ''
   ‘Introduction_R.Rmd’ using ‘UTF-8’ 
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection timed out
Warning in grDevices::png(tempfile()) :
  unable to open connection to X11 display ''
   ‘Outcrossing_Populations.Rmd’ using ‘UTF-8’ 
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection refused
Warning in grDevices::png(tempfile()) :
  unable to open connection to X11 display ''
   ‘Overview.Rmd’ using ‘UTF-8’ 
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection timed out
Warning in grDevices::png(tempfile()) :
  unable to open connection to X11 display ''
   ‘Tutorial_Onemap_complete_version.Rnw’ using ‘UTF-8’ 
   ‘Tutorial_Onemap_reduced_version.Rnw’ using ‘UTF-8’ 
** testing if installed package can be loaded
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection refused
connect pc-lsb-413a.biology.mcmaster.ca port 6000: Connection timed out
Warning in fun(libname, pkgname) :
  couldn't connect to display "localhost:10.0"
Warning: loading Rplot failed
* DONE (onemap)
```
File filtered following the GATK best practices for SNP (need to order more the file)
```
/4/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex/bwa_results/Phred20/all/GVCF/noDupliMark/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only.recode.vcf
```
With marker ID (see Hetmapps part to see how it was done)
```
/4/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/output/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf

/usr/local/tabix/bgzip /4/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/output/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf
/usr/local/tabix/tabix -p vcf /4/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/output/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf.gz
```
Convert vcf into linkage type file
```R
library(onemap)
vcf2raw(input = "/4/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/output/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf.gz", output = "Hymenochirus_HardFilteredGATK.raw", cross = "outcross", parent1 = "BJE3814", parent2 = "BJE3815")

example_out <- read_onemap("~/Hymeno_fastqc/R_packages/onemap","Hymenochirus_HardFilteredGATK.raw")
```
```R
 Working...

 --Read the following data:
	Type of cross:           outcross 
	Number of individuals:   31 
	Number of markers:       3997 
	Chromosome information:  yes 
	Position information:    yes 
	Number of traits:        0 
	
> example_out
  This is an object of class 'onemap'
    Type of cross:      outcross 
    No. individuals:    31 
    No. markers:        3997 
    CHROM information:  yes 
    POS information:    yes 
    Percent genotyped:  91 

    Segregation types:
               B3.7 -->  1549
              D1.10 -->  1220
              D2.15 -->  1228

    No. traits:         0 
```
##Script to produce a global, male and female specific maps depending on the segregation pattern of the markers.
Need to separate:
- B3.7 = shared markers (ab x ab -> a 2ab b (1:2:1))
- D1.10 = female (parent1) markers (ab x aa -> a ab (1:1))
- D2.15 = male (parent2) markers (aa x ab -> a ab (1:1))

Example from [Peng et al. 2016](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4880943/):
*sex specific linkage maps were first constructed based on SNP markers that were heterozygous only in female parent (AB × AA or AB × BB, 26,918 SNPs) or male parent (AA × AB or BB × AB, 27,157 SNPs). To provide anchor markers for female-specific map and male-specific map integration, 510 SNP markers that were heterozygous in both parents (AB × AB with 1:2:1 segregation) were also included for linkage mapping.*

`script_sex_spec_out_onemap.pl`
```perl
#!/usr/bin/env perl
use strict;
use warnings;
use List::MoreUtils qw/ uniq /;
use List::Util qw(min max);

## This script will read the marker file in a onemap input format. 
# It separates the input file into 2 different onemap type files:
# - Hymenochirus_HardFilteredGATK_female.raw with markers of types: D1.10 (ab x aa -> a ab (1:1)) and B3.7 (ab x ab -> a 2ab b (1:2:1))
# - Hymenochirus_HardFilteredGATK_male.raw with markers of types: D2.15 (aa x ab -> a ab (1:1)) and B3.7 (ab x ab -> a 2ab b (1:2:1))

#Some info from http://augusto-garcia.github.io/statgen-esalq/Hapmap-and-VCF-formats-and-its-integration-with-onemap/:
#- 1st line: information about the population type, that can be denoted as “f2 backcross” for backcross populations type, or “f2 intercross” for F2, “ri self” for RIL produced by selfing, “ri sib” for RIL produced by sib mating. 
#- The second line contains the number of individuals in the progeny, the number of markers, and the number of quantitative traits. The phenotypic information may be present at the file, but it will not be used by OneMap. 
#- After these two lines, the file must have the genotypic information for each marker. The character * indicates the beginning of information of a marker, followed by the marker name.

#NOTE: the script doesn't keep the lines at the end of the complete onemap input, i.e. the information from chromosomes and positions since we don't have real chromosomes and onemap shouldn't try to use this information

#Declare the variables
my @column;


#Path to data
my $path_to_data = "/home/evanslab/Hymeno_fastqc/R_packages/onemap/";

#Inputfile

my $inputfile = $path_to_data."Hymenochirus_HardFilteredGATK.raw";
unless (open DATAINPUT, $inputfile) {
		print "Can not find the input file $inputfile.\n";
		exit;
}

#Outputfile

my $outputfile_female = $path_to_data."Hymenochirus_HardFilteredGATK_female.raw";
unless (open(OUTFILE1, ">$outputfile_female")){
	print "I can\'t write to $outputfile_female   $!\n\n";
	exit;
}

my $outputfile_male = $path_to_data."Hymenochirus_HardFilteredGATK_male.raw";
unless (open(OUTFILE2, ">$outputfile_male")){
	print "I can\'t write to $outputfile_male   $!\n\n";
	exit;
}

while (my $line=<DATAINPUT>) {
	chomp($line);
	@column=split('\t',$line);
	if($column[0] !~ /^\*/){ #look at lines which doesn't start by the character "*", i.e. the lines at the beginning of the onemap complete file
		print "column[0] ", $column[0], "\n";
		#print "column[1] ", $column[1], "\n";
		print OUTFILE1 $line,"\n";
		print OUTFILE2 $line,"\n";
	} else {
		if ($column[1] eq 'D1.10'){
			#print "column[1] ", $column[1], "\n";
			print OUTFILE1 $line,"\n";
		}
		elsif ($column[1] eq 'D2.15'){
			print OUTFILE2 $line,"\n";
		}
		elsif ($column[1] eq 'B3.7'){ #does not need an elsif here but prefer doing so
			print OUTFILE1 $line,"\n";
			print OUTFILE2 $line,"\n";
		}
	}
}

close DATAINPUT;
print "Done with input file $inputfile\n";
close OUTFILE1;
print "Done with input file $outputfile_female\n";
close OUTFILE2;
print "Done with outputfile $outputfile_male\n";
```
