#Hetmapps
##Install dependencies
###R packages 
`WGCNA` needs `GO.db`. Couldn't use the last version of `GO.db` because required higher version for a lot of other packages. If every packages have the newest version, we needed to also update the R version (us:3.2.2, needs =>3.0) which also requires updating zlib (`zlib-1.2.10`). But some issue when try to compile even with the good version of zlib. 
```
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages qtl_1.40-8.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages snow_0.4-2.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages GO.db_3.2.2.tar.gz
setenv R_LIBS $HOME/Hymeno_fastqc/R_packages/GO.db
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages WGCNA_1.49.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages flashClust_1.01-2.tar.gz
```
###Per libraries
```
tar -xvzf ~/Hymeno_fastqc/lib_perl5/Statistics-R-0.34.tar.gz
perl Makefile.PL PREFIX=~/Hymeno_fastqc/lib_perl5/Statistics-R-0.34
make
make install
```
##Run the scripts
```
perl /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/major_minor_VCF.pl -vcf /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex/bwa_results/Phred20/all/GVCF/noDupliMark/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only.recode.vcf -o output/major_minor_VCF_Hymeno.vcf

perl get_pseudotestcross_markers.pl -vcf output/major_minor_VCF_Hymeno.vcf -i ind_file -o output/Pt_markers

setenv R_LIBS $HOME/Hymeno_fastqc/R_packages
perl denovo_phase_part1.pl -b output/Pt_markers.binary -p output/Pt_markers.parents.GQ98.binary -od output
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1.pl -b output/Pt_markers.binary -p output/Pt_markers.parents.GQ98.binary -od output

reading in data, clustering and grouping. Hold on, this may take a while
Problem while running this R command:
markers = data.frame(ID=colnames(datGtclean), CHR=as.numeric(data$CHROM[data$ID %in% colnames(datGtclean)]), POS=as.numeric(data$POS[data$ID %in% colnames(datGtclean)]), LG=match(colorStaticTOM, colorOrder)-1, PHASE = rep("NA",times=dim(datGtclean)[2]))

Error:
data.frame(ID = colnames(datGtclean), CHR = as.numeric(data$CHROM[data$ID %in%  : 
  arguments imply differing number of rows: 2132, 2138
In addition: Warning message:
In data.frame(ID = colnames(datGtclean), CHR = as.numeric(data$CHROM[data$ID %in%  :
  NAs introduced by coercion
Execution halted
```
When using `-v`:
```
Error:
could not find function "flashClust"
Execution halted
```
Need to add (l.60)
```
$status->run(qq`library(flashClust)`);
```
```
evanslab:khyma-hetmapps-8e3767dc6800 % scp output/Pt_markers.parents.GQ98.binary output/Pt_markers.parents_without_Supercontig.GQ98.binary                                                              evanslab:khyma-hetmapps-8e3767dc6800 % scp output/Pt_markers.binary output/Pt_markers.without_Supercontig.binary
evanslab:khyma-hetmapps-8e3767dc6800 % sed -i 's/supercontig_//g' output/Pt_markers.parents_without_Supercontig.GQ98.binary                                                                    evanslab:khyma-hetmapps-8e3767dc6800 % sed -i 's/supercontig_//g' output/Pt_markers.without_Supercontig.binary

perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers.without_Supercontig.binary -p output/Pt_markers.parents_without_Supercontig.GQ98.binary -od output -v
```
Same error.

Try to fix the issue by adding an ID to the marker.
```
scp /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex/bwa_results/Phred20/all/GVCF/noDupliMark/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only.recode.vcf SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf
/4/evanslab/Hymenochirus_2016/GBS_data/linkage_map/bcftools-1.3.1/bcftools annotate --set-id +'%CHROM\_%POS\_%REF\_%ALT' SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf > SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF1.recode.vcf
rm -f SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf
mv SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF1.recode.vcf SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf
```
```
perl /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/major_minor_VCF.pl -vcf output/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf -o output/major_minor_VCF_Hymeno1.vcf

perl get_pseudotestcross_markers.pl -vcf output/major_minor_VCF_Hymeno1.vcf -i ind_file -o output/Pt_markers1

perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -od output -v
```
```
Problem while running this R command:
CHRvLGtop <- data.frame(LG = colnames(CHRvLG), chr1 = row.names(CHRvLG)[chr1], val1, pos2 = row.names(CHRvLG)[chr2], val2, diff=val2/val1)

Error:
data.frame(LG = colnames(CHRvLG), chr1 = row.names(CHRvLG)[chr1],  : 
  arguments imply differing number of rows: 3, 0
Execution halted
```
Silence the part dealing with example of chromosomes and seems to work...
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -od output -v
```
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part2.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -l output/h0.85.s50.LG.txt -od output -v
```
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ phased2mstmap.pl -b output/Pt_markers1.binary -od output -l output/phased.txt -v

Type of arg 1 to keys must be hash (not hash element) at phased2mstmap.pl line 547, near "}) "
```
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ phased2rqtl.pl -b output/Pt_markers1.binary -l output/phased.txt -c 4way -v
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ phased2rqtl.pl -b output/Pt_markers1.binary -l output/phased.txt -c BC -v
```
## Looking for optimal parameters
Need to find a cut height and minimum cluster for which at least 24 LGs are created (2n=24 for H. boettgeri see [Mezzasalma et al. 2015](http://www.sciencedirect.com/science/article/pii/S0044523115300036), need to x2 because 1 map per parent). For HetMapps's pipeline documentation: 

*If chromosome number is known, the cutting parameters that resolve twice the number
of chromosomes or more should be tested.* [denovo_phase_part1.pl]
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -od output/height_min_05 -height_min 0.5 -v
```
max number of LGs created: 14.
Try with also:
```
-height_min 0.5 -minsize_min 25

HEIGHT  SIZE    LGs     TOTAL   ASSIGNED        UNASSIGNED      FILTERED
0.8375  25      20      2132    1350 (0.63)     360 (0.17)      422 (0.20)
0.825   25      20      2132    1370 (0.64)     418 (0.20)      344 (0.16)
0.8125  25      21      2132    1354 (0.64)     473 (0.22)      305 (0.14)

```
```
-height_min 0.5 -minsize_min 20
HEIGHT  SIZE    LGs     TOTAL   ASSIGNED        UNASSIGNED      FILTERED
0.8125  20      23      2132    1364 (0.64)     428 (0.20)      340 (0.16)
0.8     20      23      2132    1352 (0.63)     474 (0.22)      306 (0.14)
0.7875  20      23      2132    1315 (0.62)     552 (0.26)      265 (0.12)
```
```
-height_min 0.5 -minsize_min 15
HEIGHT  SIZE    LGs     TOTAL   ASSIGNED        UNASSIGNED      FILTERED
0.825   15      24      2132    1388 (0.65)     340 (0.16)      404 (0.19)
0.8125  15      25      2132    1371 (0.64)     395 (0.19)      366 (0.17)
0.8     15      25      2132    1356 (0.64)     442 (0.21)      334 (0.16)
0.7875  15      25      2132    1334 (0.63)     517 (0.24)      281 (0.13)
0.775   15      25      2132    1304 (0.61)     572 (0.27)      256 (0.12)
0.762500000000001       15      24      2132    1245 (0.58)     653 (0.31)      234 (0.11)

-height_min 0.3 -minsize_min 40 (fewer LGs)

-height_min 0.3 -minsize_min 10
0.85    10      27      2132    1325 (0.62)     165 (0.08)      642 (0.30)
0.8375  10      29      2132    1354 (0.64)     231 (0.11)      547 (0.26)
0.825   10      30      2132    1362 (0.64)     273 (0.13)      497 (0.23)
0.8125  10      31      2132    1349 (0.63)     329 (0.15)      454 (0.21)
0.8     10      27      2132    1346 (0.63)     421 (0.20)      365 (0.17)
0.7875  10      27      2132    1318 (0.62)     493 (0.23)      321 (0.15)
0.775   10      28      2132    1291 (0.61)     538 (0.25)      303 (0.14)
0.762500000000001       10      28      2132    1254 (0.59)     607 (0.28)      271 (0.13)
0.750000000000001       10      26      2132    1163 (0.55)     705 (0.33)      264 (0.12)

-height_min 0.8 -minsize_min 30

-height_min 0.7 -minsize_min 30

-height_min 0.7 -minsize_min 30 -height_step 0.01 -minsize_step 20
0.82    30      18      2132    1331 (0.62)     508 (0.24)      293 (0.14)
0.78    30      18      2132    1233 (0.58)     697 (0.33)      202 (0.09)
0.77    30      18      2132    1232 (0.58)     719 (0.34)      181 (0.08)

```
Needed to edit the script for `-minsize_step` (l.62): `'height_step:f' => \$height_step,`. 
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part2_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -l output/height_min_05/best/h0.825.s15.LG.txt -od output/height_min_05/best/stepnovo2 -v

could not resolve 2 phases for linakge group 5, found 1
```
Trying using different -static_height -static_minsize.

Trying using other LG files (24 or 25 LGs). Still same issue, rerun step1 of de novo phase reducing minsize... Still an issue with at least 1 LG to resolve the phases...

```
-static_height [-height_min] -static_minsize [-minsize_min]
```
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part2_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -l output/height_min_05/best/h0.825.s15.LG.txt -od output/height_min_05/best/stepnovo2 -v -static_height 0.5 -static_minsize 15

Problem while running this R command:
for (i in 6:dim(data.parents)[2]) {
                                markers.overlap[colnames(data.parents[i])] <- factor(data.parents[,i][data.parents$ID %in% markers.overlap$ID],levels=c('0','1'))
                                mtable <- table(paste(markers.overlap$LG, markers.overlap$STATIC_NUM), markers.overlap[colnames(data.parents[i])][,1])
                                GENOvPARENT[paste(colnames(data.parents[i]),".0",sep="")] <- c(mtable[1,1],mtable[2,1])
                                GENOvPARENT[paste(colnames(data.parents[i]),".1",sep="")] <- c(mtable[1,2],mtable[2,2])
                        }

Error:
`[<-.data.frame`(`*tmp*`, paste(colnames(data.parents[i]), ".0",  :
  replacement has 2 rows, data has 3
Calls: [<- -> [<-.data.frame
Execution halted

perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part2_modif.pl -b output/Pt_markers1.binary -l output/height_min_05/best/h0.825.s15.LG.txt -p output/Pt_markers1.parents.GQ98.binary -od output/height_min_05/best/stepnovo2 -static_height 0.825 -static_minsize 15
Check GENO: LG PHASE 1 1 1 2 1 2
Check: 0 1 1 1 4 103 1 2 1 42
Check GENO: LG PHASE 1 2 1 2 2 2
Check: 0 1 2 1 11 74 2 2 22 3
Check GENO: LG PHASE 1 3 1 2 3 2
Check: 0 1 3 1 4 65 3 2 0 45
Check GENO: LG PHASE 1 4 1 2 4 2
Check: 0 1 4 1 2 80 4 2 0 22
could not resolve 2 phases for linakge group 5, found 1
```
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -od output/decrease_minsize -height_min 0.5 -minsize_min 8 -v

0.725000000000001       8       24      2132    1066 (0.50)     843 (0.40)      223 (0.10)
0.8375  8       33      2132    1322 (0.62)     196 (0.09)      614 (0.29)

perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part2_modif.pl -b output/Pt_markers1.binary -l output/decrease_minsize/h0.725000000000001.s8.LG.txt -p output/Pt_markers1.parents.GQ98.binary -od output/decrease_minsize/stepnovo2 -static_height 0.725000000000001 -static_minsize 5 -v

perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -od output/decrease_minsize -height_min 0.7 -minsize_min 8 -minsize_step 20 -minsize_max 50 -v 
```
The "optimal" values find by looking in the lg_summary.txt from denovo_phase_part1.pl (h0 and so corresponding to the correct or bigger number of LGs). But for the part 2, the default values, or from step1, or other values hqven't succesfully ussed to make the step2 running. When we obtain 2 phases for some LGs, it is not the case for others, leading to the stop of the script.
