#Hetmapps
##Install dependencies
###R packages 
`WGCNA` needs `GO.db`. Couldn't use the last version of `GO.db` because required higher version for a lot of other packages. If every packages have the newest version, we needed to also update the R version (us:3.2.2, needs =>3.0) which also requires updating zlib (`zlib-1.2.10`). But some issue when try to compile even with the good version of zlib. 
```
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages qtl_1.40-8.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages snow_0.4-2.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages GO.db_3.2.2.tar.gz
setenv R_LIBS $HOME/Hymeno_fastqc/R_packages/GO.db
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages WGCNA_1.49.tar.gz
R CMD INSTALL --library=/home/evanslab/Hymeno_fastqc/R_packages flashClust_1.01-2.tar.gz
```
###Per libraries
```
tar -xvzf ~/Hymeno_fastqc/lib_perl5/Statistics-R-0.34.tar.gz
perl Makefile.PL PREFIX=~/Hymeno_fastqc/lib_perl5/Statistics-R-0.34
make
make install
```
##Run the scripts
```
perl /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/major_minor_VCF.pl -vcf /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex/bwa_results/Phred20/all/GVCF/noDupliMark/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only.recode.vcf -o output/major_minor_VCF_Hymeno.vcf

perl get_pseudotestcross_markers.pl -vcf output/major_minor_VCF_Hymeno.vcf -i ind_file -o output/Pt_markers

setenv R_LIBS $HOME/Hymeno_fastqc/R_packages
perl denovo_phase_part1.pl -b output/Pt_markers.binary -p output/Pt_markers.parents.GQ98.binary -od output
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1.pl -b output/Pt_markers.binary -p output/Pt_markers.parents.GQ98.binary -od output

reading in data, clustering and grouping. Hold on, this may take a while
Problem while running this R command:
markers = data.frame(ID=colnames(datGtclean), CHR=as.numeric(data$CHROM[data$ID %in% colnames(datGtclean)]), POS=as.numeric(data$POS[data$ID %in% colnames(datGtclean)]), LG=match(colorStaticTOM, colorOrder)-1, PHASE = rep("NA",times=dim(datGtclean)[2]))

Error:
data.frame(ID = colnames(datGtclean), CHR = as.numeric(data$CHROM[data$ID %in%  : 
  arguments imply differing number of rows: 2132, 2138
In addition: Warning message:
In data.frame(ID = colnames(datGtclean), CHR = as.numeric(data$CHROM[data$ID %in%  :
  NAs introduced by coercion
Execution halted
```
When using `-v`:
```
Error:
could not find function "flashClust"
Execution halted
```
Need to add (l.60)
```
$status->run(qq`library(flashClust)`);
```
```
evanslab:khyma-hetmapps-8e3767dc6800 % scp output/Pt_markers.parents.GQ98.binary output/Pt_markers.parents_without_Supercontig.GQ98.binary                                                              evanslab:khyma-hetmapps-8e3767dc6800 % scp output/Pt_markers.binary output/Pt_markers.without_Supercontig.binary
evanslab:khyma-hetmapps-8e3767dc6800 % sed -i 's/supercontig_//g' output/Pt_markers.parents_without_Supercontig.GQ98.binary                                                                    evanslab:khyma-hetmapps-8e3767dc6800 % sed -i 's/supercontig_//g' output/Pt_markers.without_Supercontig.binary

perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers.without_Supercontig.binary -p output/Pt_markers.parents_without_Supercontig.GQ98.binary -od output -v
```
Same error.

Try to fix the issue by adding an ID to the marker.
```
scp /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex/bwa_results/Phred20/all/GVCF/noDupliMark/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only.recode.vcf SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf
/4/evanslab/Hymenochirus_2016/GBS_data/linkage_map/bcftools-1.3.1/bcftools annotate --set-id +'%CHROM\_%POS\_%REF\_%ALT' SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf > SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF1.recode.vcf
rm -f SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf
mv SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF1.recode.vcf SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf
```
```
perl /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/linkage_map/khyma-hetmapps-8e3767dc6800/major_minor_VCF.pl -vcf output/SOAP_Chimerical_recalibrated_round1_allsites_Phred20_HardFilteringSNP_filtered_family_only_REF.recode.vcf -o output/major_minor_VCF_Hymeno1.vcf

perl get_pseudotestcross_markers.pl -vcf output/major_minor_VCF_Hymeno1.vcf -i ind_file -o output/Pt_markers1

perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -od output -v
```
```
Problem while running this R command:
CHRvLGtop <- data.frame(LG = colnames(CHRvLG), chr1 = row.names(CHRvLG)[chr1], val1, pos2 = row.names(CHRvLG)[chr2], val2, diff=val2/val1)

Error:
data.frame(LG = colnames(CHRvLG), chr1 = row.names(CHRvLG)[chr1],  : 
  arguments imply differing number of rows: 3, 0
Execution halted
```
Silence the part dealing with example of chromosomes and seems to work...
```
perl -I /home/evanslab/Hymeno_fastqc/lib_perl5/Statistics-R-0.34/share/perl5/ -I /home/evanslab/Hymeno_fastqc/lib_perl5/Regexp-Common-2016060801/share/perl5/ denovo_phase_part1_modif.pl -b output/Pt_markers1.binary -p output/Pt_markers1.parents.GQ98.binary -od output -v
```
