## Demultiplexing
 
###RADpools 
`RADpools --version`:
Getopt::Long::GetOptions version 2.38; Perl version 5.10.1
```
/usr/local/RADtools/RADpools -v [verbose] -f [If a MID sequence in a read contains an error, reads will be accepted and assigned to the nearest pool] -i [file] -d [output directory name, match sample tags file name]  -m [max_processes, default:1, set to 2] -e [restriction enzyme seq] -s -o [output_fastq]

perl -I /home/evanslab/RADtools-master /home/evanslab/RADtools-master/RADpools -v -f -i /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -d Hymenochirus_barcode_RADpools -m 2 -s -o

perl -I /home/evanslab/RADtools-master /home/evanslab/RADtools-master/RADpools -v -f -i /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -d Hymenochirus_barcode_RADpools -m 2 -s -o
```
`perl -I /home/evanslab/RADtools-master` let us to include `/home/evanslab/RADtools-master` where `RADtools.pm*` is located to the beginning of @INC for this specific execution of the script. 

Results
```
258453 records loaded, but discarded
0 records not in FASTQ format,
0 records not in RAD format,
158544888 records not matching restriction enzyme site,
519231 records not matching any listed P1 MIDs,
0 records not matching any listed P2 MIDs, and
0 records with high-quality sequences less than 85 bp long
Done

```
Check that no a space/tab pb: remake the barcode file replacing tab by space (even if should be ok but tab and space look like the same on emacs and vi) 
```
perl -I /home/evanslab/RADtools-master /home/evanslab/RADtools-master/RADpools -v -f -i /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -d Hymenochirus_barcode_RADpools_space -m 2 -s -o

258453 records loaded, but discarded
0 records not in FASTQ format,
0 records not in RAD format,
158544888 records not matching restriction enzyme site,
519231 records not matching any listed P1 MIDs,
0 records not matching any listed P2 MIDs, and
0 records with high-quality sequences less than 85 bp long
Done
```
Check that it is not a problem due to multiple barcodes for the same sample. Make each sample name unique (i.e. add `_1`, `_2`... when the same sample appear multiple times. 
```
perl -I /home/evanslab/RADtools-master /home/evanslab/RADtools-master/RADpools -v -f -i /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -d Hymenochirus_barcode_RADpools_space_uniq -m 2 -s -o
```
Results
```
682776 records loaded, but discarded
0 records not in FASTQ format,
0 records not in RAD format,
158544888 records not matching restriction enzyme site,
94908 records not matching any listed P1 MIDs,
0 records not matching any listed P2 MIDs, and
0 records with high-quality sequences less than 85 bp long
Done
```
```
perl -I /home/evanslab/RADtools-master /home/evanslab/RADtools-master/RADpools -v -f -i /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -d Hymenochirus_barcode_RADpools_correct.pools -m 2 -s -o
```
###STACKS: process_radtags
```
/usr/local/fastqc/fastqc -o /home/evanslab/Hymeno_fastqc/GBS_analysis/ /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq
```
No overrepresented sequences identified.
```
/usr/local/bin/process_radtags -f /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -b /home/evanslab/Hymeno_fastqc/GBS_analysis/Hymenochirus_barcode -o /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex -e sbfI -t 75 -r -c -q --filter_illumina
```
Results
```
159322572 total sequences;
  0 failed Illumina filtered reads;
  18094064 ambiguous barcode drops;
  8805 low quality read drops;
  140545732 ambiguous RAD-Tag drops;
673971 retained reads.
```
with -t 70
```
159322572 total sequences;
  0 failed Illumina filtered reads;
  18094064 ambiguous barcode drops;
  8035 low quality read drops;
  140545732 ambiguous RAD-Tag drops;
674741 retained reads.
```
Sounds like people had issue with enzyme sites containing `N` or not being recognized by the software.
```
cat /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq | grep "^TCCGGAGC" | cut -c 9-16 | sort | uniq -c
...
 50 CCTGCAGG
 ...
 393 GCTGCAAA
 178 GCTGCAAC
 263 GCTGCAAG
 369 GCTGCAAT
 395 GCTGCACA
 207 GCTGCACC
 327 GCTGCACG
 219 GCTGCACT
 665 GCTGCAGA
 3161 GCTGCAGC
 879367 GCTGCAGG
 ...
 7 NCTGCAGG
 ...
```
Variations from the restriction site in their [paper](http://www.g3journal.org/content/1/3/171.full) `CCTGCAGG`.

***"Uncalled nucleotides, nonexistent barcodes, or deficient restriction enzyme cut sites can also cause Stacks to exclude reads. Stacks can correct isolated errors in the restriction cut site sequence or in the barcode if the barcode is two or more nucleotides distant in sequence space from other barcodes used in the same sequencing library. "*** ([Catchen et al. 2011](http://www.g3journal.org/content/1/3/171.full))

Let's use the option `--disable_rad_check` for not checking the restriction site.
```
/usr/local/bin/process_radtags -f /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -b /home/evanslab/Hymeno_fastqc/GBS_analysis/Hymenochirus_barcode -o /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex -e sbfI -t 70 -r -c -q --filter_illumina --disable_rad_check
```
Results
```
159322572 total sequences;
  0 failed Illumina filtered reads;
  18094064 ambiguous barcode drops;
  1382003 low quality read drops;
  0 ambiguous RAD-Tag drops;
139846505 retained reads.
```
```
/usr/local/bin/process_radtags -f /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -b /home/evanslab/Hymeno_fastqc/GBS_analysis/Hymenochirus_barcode_corr -o /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex -e sbfI -t 70 -r -c -q --filter_illumina
```

## [Checking sequences quality](https://github.com/caro46/Hymenochirus/blob/master/Starting.Rmd)
- Fastqc
- Trimmomatic
- Quake (?)

## GATK + vcftools
- linkage stats: `./vcftools --vcf input_data.vcf --hap-r2 --ld-window-bp 50000 --out ld_window_50000`.

## Checking sex
- SNPhylo / SNPrelate (R) using vcf file ? -> build a phylo to check males all together and females separated (if not: check sex on pictures + individual under microscope)

## Plink 
- vcftools `./vcftools --vcf input_data.vcf --plink --chr 1 --out output_in_plink` (output `.map` `.ped`).
- with R: Manhattan plot

##JOINMAP?
Compare linkage dist. between male and female?

## [R OneMap](http://augusto-garcia.github.io/statgen-esalq/Hapmap-and-VCF-formats-and-its-integration-with-onemap/)
