## Demultiplexing
 
`RADpools --version`:
Getopt::Long::GetOptions version 2.38; Perl version 5.10.1
```
/usr/local/RADtools/RADpools -v [verbose] -f [If a MID sequence in a read contains an error, reads will be accepted and assigned to the nearest pool] -i [file] -d [output directory name, match sample tags file name]  -m [max_processes, default:1, set to 2] -e [restriction enzyme seq] -s -o [output_fastq]

perl -I /home/evanslab/RADtools-master /home/evanslab/RADtools-master/RADpools -v -f -i /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -d /home/evanslab/Hymeno_fastqc/GBS_analysis/Hymenochirus_barcode -m 2 -s -o
```
`perl -I /home/evanslab/RADtools-master` let us to include `/home/evanslab/RADtools-master` where `RADtools.pm*` is located to the beginning of @INC for this specific execution of the script. 

OR
```
/usr/local/fastqc/fastqc -o /home/evanslab/Hymeno_fastqc/GBS_analysis/ /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq
```
No overrepresented sequences identified.
```
/usr/local/bin/process_radtags -f /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -b /home/evanslab/Hymeno_fastqc/GBS_analysis/Hymenochirus_barcode -o /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex -e sbfI -t 75 -r -c -q --filter_illumina
```
Results
```
159322572 total sequences;
  0 failed Illumina filtered reads;
  18094064 ambiguous barcode drops;
  8805 low quality read drops;
  140545732 ambiguous RAD-Tag drops;
673971 retained reads.
```
with -t 70
```
159322572 total sequences;
  0 failed Illumina filtered reads;
  18094064 ambiguous barcode drops;
  8035 low quality read drops;
  140545732 ambiguous RAD-Tag drops;
674741 retained reads.
```
Sounds like people had issue with enzyme sites containing `N` or not being recognized by the software.
```
cat /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq | grep "^TCCGGAGCGCTGCAGG" | cut -c 9-16 | sort | uniq -c
 879367 GCTGCAGG
```
Checking using different barcode, we always obtained `GCTGCAGG` which is only 1 bp difference from the restriction site in their [paper](http://www.g3journal.org/content/1/3/171.full) `CCTGCAGG`.

***"Uncalled nucleotides, nonexistent barcodes, or deficient restriction enzyme cut sites can also cause Stacks to exclude reads. Stacks can correct isolated errors in the restriction cut site sequence or in the barcode if the barcode is two or more nucleotides distant in sequence space from other barcodes used in the same sequencing library. "*** ([Catchen et al. 2011](http://www.g3journal.org/content/1/3/171.full))

Let's use the option `--disable_rad_check` for not checking the restriction site.
```
/usr/local/bin/process_radtags -f /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/563_S7_L007_R1_001.fastq -b /home/evanslab/Hymeno_fastqc/GBS_analysis/Hymenochirus_barcode -o /net/infofile4-inside/volume1/scratch/evanslab/Hymenochirus_2016/GBS_data/process_radtags_demultiplex -e sbfI -t 70 -r -c -q --filter_illumina --disable_rad_check
```
Results
```
159322572 total sequences;
  0 failed Illumina filtered reads;
  18094064 ambiguous barcode drops;
  1382003 low quality read drops;
  0 ambiguous RAD-Tag drops;
139846505 retained reads.
```

## [Checking sequences quality](https://github.com/caro46/Hymenochirus/blob/master/Starting.Rmd)
- Fastqc
- Trimmomatic
- Quake (?)

## GATK + vcftools
- linkage stats: `./vcftools --vcf input_data.vcf --hap-r2 --ld-window-bp 50000 --out ld_window_50000`.

## Checking sex
- SNPhylo / SNPrelate (R) using vcf file ? -> build a phylo to check males all together and females separated (if not: check sex on pictures + individual under microscope)

## Plink 
- vcftools `./vcftools --vcf input_data.vcf --plink --chr 1 --out output_in_plink` (output `.map` `.ped`).
- with R: Manhattan plot

##JOINMAP?
Compare linkage dist. between male and female?

## [R OneMap](http://augusto-garcia.github.io/statgen-esalq/Hapmap-and-VCF-formats-and-its-integration-with-onemap/)
