## Detection before assembly
Use de novo assembly to build repeat libraries from reads files: Tedna (difficult to install on Info) and [RepARK](https://github.com/PhKoch/RepARK) (which is young, 2014, but seems to run easily using Velvet and Jellyfish on Info):

### Building TE libraries with RepARK
```
/home/evanslab/Hymeno_fastqc/RepArk_analysis/RepARK-master/RepARK.pl -l /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz -l /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz
```
Results:

BJE3815: Theshold used: 138 (`Check the directory RepARK_working for results.` See in it: `repeat_lib.fasta@`)

BJE3814: Theshold used: 92

### Identifying TE classes with [`TEclass-2.1.3`](http://www.compgen.uni-muenster.de/tools/teclass/index.hbi?lang=en) (classifiers from RepBase 18.10)
```
/home/evanslab/Hymeno_fastqc/RepArk_analysis/TEclass-2.1.3/TEclassTest.pl -r /home/evanslab/Hymeno_fastqc/RepArk_analysis/repeat_BJE3814/RepARK_working/repeat_lib.fasta -o /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TE_class_BJE3814

/home/evanslab/Hymeno_fastqc/RepArk_analysis/TEclass-2.1.3/TEclassTest.pl -r /home/evanslab/Hymeno_fastqc/RepArk_analysis/repeat_BJE3815/RepARK_working/repeat_lib.fasta -o /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/
```
#### Results for BJE3814
.stat:
```
Repeat statistics:
DNA transposons: 86784
LTRs:  46282
LINEs: 8506
SINEs: 15300
Unclear:  54341

Total: 211213
```
.lib (add TE class in Fasta label of the sequences of identified TE)
```
>NODE_1000005_length_41_cov_4.975610|TEclass result: LTR|forward
TCCCATAATAGTTTTATGACATCAGTAAGATTTTAGTCATAGTTGGAGAGACATTGAGGTGAGTGGACC
>NODE_100000_length_59_cov_2.932203|TEclass result: Retro|forward
TAGGTGGTTGTGGGTATATCTAGAGTTATTATACTAGGTGGTTGTGGGTATATCTAGAGAGGTAACACTGGGTGGTTGTG
GGTATAT
...
```
#### Results for BJE3815
```
Repeat statistics:
DNA transposons: 56381
LTRs:  28799
LINEs: 5478
SINEs: 9257
Unclear:  33655

Total: 133570
```
## Detection after assembly

###RepeatMasker (`RepeatMasker version open-3.2.6`)

```
RepeatMasker [-options] <seqfiles(s) in fasta format>
/usr/local/RepeatMasker/RepeatMasker -dir /home/evanslab/Hymeno_fastqc/repeatmasker_analysis  -a /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3815-8.fa

/usr/local/RepeatMasker/RepeatMasker -dir /home/evanslab/Hymeno_fastqc/repeatmasker_analysis -species "xenopus genus" -a /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3815-8.fa
```
When we will obtain the results: add our library made previously with RepARK and TEclass, with:
```
/usr/local/RepeatMasker/RepeatMasker -lib /home/evanslab/Hymeno_fastqc/RepArk_analysis/repeat_BJE3814 -dir /home/evanslab/Hymeno_fastqc/repeatmasker_analysis  -a /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/abyss/BJE3814-8.fa
```
#### On scaffolds identified as caracteristic of one sex
```
/usr/local/RepeatMasker/RepeatMasker -lib /home/evanslab/Hymeno_fastqc/RepArk_analysis/repeat_BJE3814/repeat_lib.fasta_1458332765/repeat_lib.fasta.lib -dir /home/evanslab/Hymeno_fastqc/repeatmasker_analysis/scaff_sup_fem_depthRound1  -a /home/evanslab/Hymeno_fastqc/repeatmasker_analysis/scaff_sup_fem_depthRound1/scaffolds_sup_fem_caract_depth_round1.fa
```
##### Blast the results
```
blastn -evalue 1e-20 -query /work/ben/2016_Hymenochirus/BJE3814/blast/repeat_masker_blast/scaffolds_sup_fem_caract_depth_round1.fa.masked -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_masked_blastable -out /work/ben/2016_Hymenochirus/BJE3814/blast/repeat_masker_blast/scaffolds_sup_fem_caract_depth_round1_masked_blast_trop9 -outfmt 6 -max_target_seqs 1
```
> No alignment length > 60bp (use e-value of 10^-20 and 10^-5)... > suggest our potential "female specific regions" very repetitive. See what happens when we mask the whole genome and map read against it.


## Comparison with *X. tropicalis*

Do it on trop9 not masked:

makeblastdb -in /work/ben/2016_Hymenochirus/xenTro9/Xtropicalis_v9.fa -dbtype nucl -title xenBase_xenTro9 -out /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_NOTmasked_blastable

blastn -evalue 1e-20 -query /work/ben/2016_Hymenochirus/ -db /work/ben/2016_Hymenochirus/xenTro9/xenTro9_genome_NOTmasked_blastable -out /work/ben/2016_Hymenochirus/ -outfmt 6 -max_target_seqs 1
