#### Size of the subdirectories within a directory
`du -h --max-depth=1`
#### Size of multiple directories and total
```
du -csh ostechnix /home/sk/
```
#### Number of files in a directory
`ls -1 | wc -l` within the directory.
#### `.tar` files
- How to compress into a `.tar` file

`tar -cvf Sample_BenEvansBJE3652_180bp_Library.tar Sample_BenEvansBJE3652_180bp_Library`

- See the files contained in a `.tar` file

`tar -tf BJE3652_6KB_Data_first_lane.tar`

- See the files contained in a `.tar.gz` file

`tar -ztvf file.tar.gz`

- Comparing `.tar.gz` with the corresponding folder

`tar --compare --file=2016_Hymenochirus_SOAP_assemblies.tar.gz`

- Extracting only one file from a `.tar.gz`

`tar -zxvf config.tar.gz etc/default/sysstat` for a file called `sysstat`

- Adding files to a `tar.gz` (from [askunbuntu](https://askubuntu.com/questions/267344/how-can-i-update-a-tar-gz-file))

1. Extract .tar file from .tar.gz file: `gunzip filename.tar.gz`
2. Update uncompressed .tar file with tar -u command: `tar -uf filename.tar new_file`
3. Compress the updated .tar file: `gzip filename.tar`


#### Extract one seq from a big fasta file 
```
samtools faidx SOAP_Hymeno_genome_scafSeq_supercontigs.fa
samtools faidx SOAP_Hymeno_genome_scafSeq_supercontigs.fa supercontig_7 >SOAP_Hymeno_genome_scafSeq_supercontigs_7only.fa
```
### Perl libraries
```
cpan App::cpanminus
cpan List::MoreUtils
```
### Counting number of files which match patterns in their names within a directory
```
find . -maxdepth 1 -name "*L006*_R1*_trim_paired.fastq.gz"| wc -l
```
### Replace tab in a file
```
sed -i 's/\t/ /g' Hymenochirus_barcode_RADpools_no_overhang.pools
```
### Local installation of gcc
Can be downloaded from [here](ftp://ftp.gnu.org/gnu/gcc/gcc-6.2.0/). If only version `g++ (GCC) 4.4.7` on Info, need to do following steps:
```
 tar xzf gcc-6.2.0.tar.gz
 cd gcc-6.2.0
 ./contrib/download_prerequisites
 cd ..
 mkdir objdir
 cd objdir
 uname -m
 $PWD/../gcc-6.2.0/configure --prefix=/home/evanslab/Hymeno_fastqc/gcc-6.2.0 --enable-languages=c,c++,fortran,go --disable-multilib
 make
 make install
```
### Print a sequence with `awk`
```
awk -v seq="scaffold348337" -v RS='>' '$1 == seq {print RS $0}' Hymenochirus_putative_sex_linked_opt_REPARKmasked.fa.masked >scaffold348337.fa
```
### Select sequences of a minimal size with `awk`
```
awk '!/^>/ { next } { getline seq } length(seq) >= 500 { print $0 "\n" seq }' /home/caroline/hymeno/MUMMER_analysis/BJE3815-8.fa >/home/caroline/hymeno/MUMMER_analysis/BJE3815-8_500bp.fa
zcat /4/caroline/2017_Pipoidea_Hiseq/Assemblies/Rhynophrinus/abyss/CSL6209-8.fa.gz | awk '!/^>/ { next } { getline seq } length(seq) >= 500 { print $0 "\n" seq }' - >/4/caroline/2017_Pipoidea_Hiseq/Assemblies/Rhynophrinus/abyss/CSL6209-8_500bp.fa
awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' /4/caroline/2017_Pipoidea_Hiseq/Assemblies/SOAP_pipa_genome_chimerical_43mers.scafSeq >/4/caroline/2017_Pipoidea_Hiseq/Assemblies/SOAP_pipa_genome_chimerical_43mers_one_line.fa
awk '!/^>/ { next } { getline seq } length(seq) >= 500 { print $0 "\n" seq }' /4/caroline/2017_Pipoidea_Hiseq/Assemblies/SOAP_pipa_genome_chimerical_43mers_one_line.fa >/4/caroline/2017_Pipoidea_Hiseq/Assemblies/SOAP_pipa_genome_chimerical_43mers_500bp.fa
```
### Force detached screen from other ssh
```
screen -d -r [name]
```
### To move stuff from /work to /freezer on sharcnet
From a loggin node (ex.: wobbie)
```
ssh cauretc@wobbie.sharcnet.ca
ssh wob101
ssh cauretc@dtn.sharcnet.ca #data transfer node, better when want to move big data

#From /work/cauretc
mv 2016_Hymenochirus/2016_Hymenochirus_SOAP_assemblies.tar.gz /freezer/cauretc
```
To see the `freezer` content: directly `ls /freezer/cauretc` after 1st loggin on `wobbie`, do not log on another `wob` like `wob101`
```
#Sept.13
ls /freezer/cauretc
2016_Hymenochirus_SOAP_assemblies.tar.gz  abyss_Rat   MIMAR_2015_2016_storage
2017_Pipoidea                             Assemblies
ls /freezer/cauretc/2017_Pipoidea/
CSL6209_bbduk_skewer.list  CSL6209_bbduk_skewer.tar.gz
```
### Soft to hard mask
```
tr '[:lower:]' 'N' </4/caroline/laevis_genome/Xla.v91.repeatMasked.fa >/4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK.fa
sed 's/>SNNNNNNN/Scaffold/' /4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK.fa >/4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK1.fa
sed 's/>NNN/>chr/' /4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK1.fa >/4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK2.fa
rm -f /4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK1.fa
rm -f /4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK.fa
mv /4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK2.fa /4/caroline/laevis_genome/Xla.v91_repeatMasked_HARD_MASK.fa
``` 
