Following is what it needs to be done when we will have *Hymenochirus* sequences (mother BJE3814 + father BJE3815).

# Before assembling

## Quality of the sequences

```
fastqc -o [location/name of the output] [location/name of the input] 
fastqc -o /home/evanslab/Hymeno_fastqc /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/*.fastq.gz
```
Globally: quality OK, 1 kmer fail: R1-BJE3814 -> check if still the same pb after trimmomatic (pics at the beginning = adapters + increase of the quantity of kmers from the beginning to the end of the sequences...).

## Run trimmomatic on the sequences 
We use the same script as for the [tetraploid project](https://github.com/caro46/Tetraploid_project/blob/master/quality_tests.Rmd) (phred quality detected = phred33).
+ check the quality again.

##Use Quake or Reptile 

Estimate the genome size using k-mers (size =  total length of used reads divided by sequencing depth) + TE proportion

G = (N × (L – K + 1) – B)/D (from [Lamichhaney et al. 2016](http://www.nature.com/ng/journal/v48/n1/fig_tab/ng.3430_SF1.html))

> Where:
>
> - G = genome size
>
> - N = total count of used reads (we can fgrep function)
>
> - L = length of used reads (can use grep function)
>
> - K = k-mer length
>
> - B = total count of low-frequency (frequency ≤1) k-mers that are probably caused by sequencing errors
>
> - D = sequencing depth

D & B can be estimated after ploting the histogram `histo` from jellyfish.

+ compare histo of female and male to check if difference in the amount of kmer counts (difference peaks size)

## Use TEDNA software ([Zytnicki et al. 2014](http://bioinformatics.oxfordjournals.org/content/30/18/2656.long)) to identify TE?

# Reference genome: *Xenopus tropicalis*
We will try to assemble the genome of *Hymenochirus* using as a reference another diploid species *X. tropicalis* which as an annotated genome (xenTro7.fa.masked.gz from [UCSC Genome Bioinformatics](http://hgdownload.cse.ucsc.edu/downloads.html#xentro)). 

References:

Lamichhaney et al. Structural genomic changes underlie alternative reproductive strategies in the ruff (Philomachus pugnax). (2016) Nature Genetics 48, 84–88. doi:10.1038/ng.3430
