Following is what it needs to be done when we will have *Hymenochirus* sequences (mother BJE3814 + father BJE3815).

# Before assembling

## Quality of the sequences

```
fastqc -o [location/name of the output] [location/name of the input] 
fastqc -o /home/evanslab/Hymeno_fastqc /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/*.fastq.gz
```
Globally: quality OK, 1 kmer fail: R1-BJE3814 -> check if still the same pb after trimmomatic (pics at the beginning = adapters + increase of the quantity of kmers from the beginning to the end of the sequences...). -> OK after trimmomatic

## Run trimmomatic on the sequences 
We use the same script as for the [tetraploid project](https://github.com/caro46/Tetraploid_project/blob/master/quality_tests.Rmd) (phred quality detected = phred33).
+ check the quality again: BJE3815_R1 has now a warning for kmers -> real kmers (check using `Quake`).

##Use Quake or Reptile 
```
zcat /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_*_trim_paired.fastq.gz | /home/evanslab/tetra_project/jellyfish-2.2.4/bin/jellyfish count /dev/fd/0 -m 19 -s 100M -t 16 -C -o BJE3814_jelly_count_19mers

zcat /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_*_trim_paired.fastq.gz | /home/evanslab/tetra_project/jellyfish-2.2.4/bin/jellyfish count /dev/fd/0 -m 19 -s 100M -t 16 -C -o BJE3815_jelly_count_19mers
```
```
/home/evanslab/tetra_project/jellyfish-2.2.4/bin/jellyfish dump -c -t BJE3814_jelly_count_19mers -o BJE3814_jelly_dump_19mers
/usr/local/quake/bin/cov_model.py --int /home/evanslab/Hymeno_fastqc/jellyfish_Hymeno/BJE3814_jelly_dump_19mers
```
Cutoff = 4 for both individuals

```R
histoM=read.table("kmers.hist",sep="\t")
histoF=read.table("kmers_female.hist",sep="\t")
pdf('19mers_distribution_quake_HymenochirusFemaleMale_5_200_bis.pdf')
plot(histoF[5:200,],type="l",col="pink",ylab="count",xlab="coverage",main="19-mers distribution (Hymenochirus Female & Male)")
lines (histoM[5:200,], col="blue")
legend('topright',c("Female BJE3814","Male BJE3815"),lty=c(1,1),lwd=c(2.5,2.5),col=c("pink","blue"))
dev.off()
```

To check the efficiency of Quake and particularly `correct`, we redo the plot of the kmers distribution
```
zcat /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_*_trim_paired.cor.fastq.gz | /home/evanslab/tetra_project/jellyfish-2.2.4/bin/jellyfish count /dev/fd/0 -m 19 -s 100M -t 16 -C -o BJE3814_jelly_count_19mers_after_quake_correct

/home/evanslab/tetra_project/jellyfish-2.2.4/bin/jellyfish dump -c -t BJE3814_jelly_count_19mers_after_quake_correct -o BJE3814_jelly_dump_19mers_after_quake

/usr/local/quake/bin/cov_model.py --int /home/evanslab/Hymeno_fastqc/quake_correct_essai2/BJE3814_jelly_dump_19mers_after_quake

```

Estimate the genome size using k-mers (size =  total length of used reads divided by sequencing depth) + TE proportion

G = (N × (L – K + 1) – B)/D (from [Lamichhaney et al. 2016](http://www.nature.com/ng/journal/v48/n1/fig_tab/ng.3430_SF1.html))

> Where:
>
> - G = genome size
>
> - N = total count of used reads (we can fgrep function)
>
> - L = length of used reads (can use grep function)
>
> - K = k-mer length
>
> - B = total count of low-frequency (frequency ≤1) k-mers that are probably caused by sequencing errors
>
> - D = sequencing depth

D & B can be estimated after ploting the histogram `histo` from jellyfish.

For the female: G = (N × (L – 19 + 1) – B)/D

Estimates of trusted kmers (with `correct`):

- BJE3814: 2520143820 trusted kmers
AT% = 0.610611
- BJE3815: 2527944394 trusted kmers
 AT% = 0.610568

Number of total kmers (counting the number of lines in the `dump` file using `wc -l BJE3815_jelly_dump_19mers`):

- BJE3814: 3912234554 

- BJE3815: 4077734314 

Number of false kmers:

- BJE3814: 3912234554 - 2520143820 = 1392090734

- BJE3815: 4077734314 - 2527944394 = 1549789920
 
Number of used reads (from FastQC files):

- BJE3814: 945915602

- BJE3815: 961651938

BJE3814 - reads (python script)

- Sequences found:  73224891.0

- Avg length:  241.415580202

**+** compare histo of female and male to check if difference in the amount of kmer counts (difference peaks size). For the female, we obtain a higher peak around ~cov 50.

## Use TEDNA software ([Zytnicki et al. 2014](http://bioinformatics.oxfordjournals.org/content/30/18/2656.long)) to identify TE?

```
./configure --prefix=$HOME
make
make check
make install
```

## Use Cortex software to assemble and compare the sequences of male and female
See [Cortex website](http://cortexassembler.sourceforge.net/index.html), the main [publication](http://www.nature.com/ng/journal/v44/n2/pdf/ng.1028.pdf) and the [manual](http://cortexassembler.sourceforge.net/cortex_var_user_manual.pdf)
```
emacs INDEX
BJE3814 . /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz
BJE3815 . /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S3_L003_R1_001_trim_paired.cor.fastq.gz /net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S3_L003_R2_001_trim_paired.cor.fastq.gz
```
```
emacs paired_end_list1
/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R1_001_trim_paired.cor.fastq.gz
/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S3_L003_R1_001_trim_paired.cor.fastq.gz

emacs paired_end_list2
/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3814_S3_L003_R2_001_trim_paired.cor.fastq.gz
/net/infofile2/2/scratch/evanslab_backups/Hymenochirus_2016/TCAG_20160129/160106_E00389_0017_AHK3LJCCXX/BJE3815_S3_L003_R2_001_trim_paired.cor.fastq.gz

./cortex_var_31_c1 --kmer_size 31 --mem_height 17 --mem_width 100 --pe_list paired_end_list1,paired_end_list2 --dump_binary /path
./cortex_var_31_c1 --kmer_size 31 --mem_height 25 --mem_width 100 --colour_list colour1.txt --detect_bubbles1 *0/0 --output_bubbles1 vars_between_cols0_and_others
```
# Reference genome: *Xenopus tropicalis*
We will try to assemble the genome of *Hymenochirus* using as a reference another diploid species *X. tropicalis* which has an annotated genome (xenTro7.fa.masked.gz from [UCSC Genome Bioinformatics](http://hgdownload.cse.ucsc.edu/downloads.html#xentro)). 

References:

Lamichhaney et al. Structural genomic changes underlie alternative reproductive strategies in the ruff (Philomachus pugnax). (2016) Nature Genetics 48, 84–88. doi:10.1038/ng.3430
