## Permutation tests
[Gamble et al. 2017](http://ac.els-cdn.com/S096098221730711X/1-s2.0-S096098221730711X-main.pdf?_tid=96291c02-972e-11e7-af1e-00000aacb35f&acdnat=1505161380_c7581d4fa26a658549dbc193e697af07) did permutations tests by resampling the sex of the individuals.

See [Jonathan's and Ben's course](https://mac-theobio.github.io/QMEE/permutation_examples.html) for some R commands.

The idea: comparing the number of sex linked sites we obtain from our family data with the distribution of "sex linked markers" obtained randomly shuffling the sex of the individuals (shuffling the sex of the offspring). Should do it for "only 1 sex" site (I checked on a few with the HiSeq data and the de novo assemblies to see if the 2 set of data agrees for some it was not the case, we can see by similating a random generation of the data if our data are more likely not due to random/errors), sex specific SNP and globally.

Pratical idea: add a loop in the previous scripts to produce 100 different datasets and then plot the distribution of the number of "sex specific" sites. Should add a line corresponding to the number of XY and Zw sites. Then something like `2*mean(res>=obs)` to have a p-value. 

### Shuffling the ID files 
Using `slice` function from `dplyr` to select the sex of the offspring. Then using `sample` to shuffle the sex of the offspring. The following script will produce new random ID files containing the real ID of the parents and the newly assigned sex for each offspring:
```R
setwd("~/Documents/caroline/linkage_try/qtl")
library("dplyr")

#data
id <- read.table("Sample_ID.txt",sep="\t",h=T)
id

#select offspring only
id_offspring <- slice(id, 3:nrow(id))
#select parents only
id_parents <- slice(id, 1:2)

for (i in 1:1000) {
	id_1 <- transform( id_offspring, Sex = sample(Sex) )
	id_1 <- rbind(id_parents,id_1)
	write.table(id_1, file = paste("./random_ID_Pipa/Sample_ID_random_",i,".txt", sep=""), sep = "\t",quote=F,row.names =F)
}
```
We only shuffle the ID for the offspring since their genotypes depend of the genotypes of the parents.

Now we have the inputs for the other scripts. We need to add a loop to them and run them again in a separate folder. Previously run R to do 1000 permutations which created 1000 files. We will try everything with only 100 files 1st (to avoid having too many intermediary files). The number of permutations matter for the accuracy of the estimation of the p-value (Gamble et al. 2017 used 100 permutations). 

Recommendations from [Whashington University](http://faculty.washington.edu/kenrice/sisg/SISG-08-06.pdf)

*With 1000 permutations the smallest possible p-value is 0.001, and the uncertainty near p = 0.05  is about Â±1%*

So with 100 permutations the smallest possible p-value should be 0.01. We should probably try with 200 permutations so smallest p-value would be 0.005 which is good. 

### Changing the scripts

The easiest would probably to just call both of them within another one

`source_simulation.pl`
```perl
#!/usr/local/perl5.24/perl-5.24.0/perl

use strict;
use warnings;

## Argument for the perl scrip
my $output_path = $ARGV[0];
my $Genotype_file = $ARGV[1];
my $pedigree = $ARGV[2];
my $Sex_linked_sites = $output_path."\/Sex_linked";
my $Polym_sites = $output_path."\/Polym_sites";
my $Invalid_sites = $output_path."\/Invalid_sites";

## C++ 
my $count = $output_path."\/count_permutation.txt";
my $scaffold_number = $output_path."\/count_scaffolds_per_SD_system.txt";

## Other variables
my $commandline;
my $status;
my $y;

##Starting the loop
for ($y = 1 ; $y <= 200 ; $y++ ) {

##Calling the 1st perl script
	$commandline= "perl script_find_sex_linked_regions ".$Genotype_file." ".$pedigree."_".$y.".txt ".$Sex_linked_sites.$y." ".$Polym_sites.$y." ".$Invalid_sites.$y;
	$status = system($commandline);

##Calling the 2nd C++ script
	$commandline= "make makefile "; #to load the library and compile
	$status = system($commandline);
	$commandline= "./main.cpp ".$Sex_linked_sites.$y." ".$Polym_sites.$y." ".$Genotype_file." ".$count.$y" ".$scaffold_number." ".$y; 
        $status = system($commandline);

##Remove the intermediary files
	$commandline= "rm -f ".$Sex_linked_sites.$y." ".$Polym_sites.$y." ".$Invalid_sites.$y;
        $status = system($commandline); 
}
## should be able to just update for each value of i the outputs from the c++ script and only have 2 final files if yes we can directly do the 1000 permutations 
## if not we should concatenate the outputs of the script and again delete the intermediary files
## not final script: need to change the script_find_sex_linked_regions and the main.cpp to have the input and output files as arguments and also see if concatenation necessary
```
We should probabbly change the `count.txt` output since we do not need the scaffold names for the simulations.

Should only contain `number_total_called_genotype \t number_XY_sites \t Yonly \t number_ZW_sites \t Wonly \t number_polymorphic_dad_sites \t number_polymorphic_mom_sites \t number_polymorphic_daughter_sites \t number_polymorphic_son_sites`. 

For both outputs we should have the `i` value at the beginning of the line. 

Need to edit the `main.cpp` with `const char* sex_linked_file = arg[0]` and similar thing for the other files. 

Sept 15: Should be good to go: only need to uncomment the non necessary output files. I'll run also the `c++` simulation on the real data to have exactly the same outputs as the simulations. In order to not have a lot of intermediary or final files, for each permutation, the results will be append at the end of the outputfiles; so should be fine to run the 1000 permutations.
They will be run from `/4/caroline/random_simulation` by 

```
perl source_simulation.pl /4/caroline/random_simulation/pipa_simulation_SL /4/caroline/Pipa_parva/Rad_seq/GATK/SOAP_Chimerical_recalibrated_allsites_Phred20_HardFilteringSNP_filtered.tab /4/caroline/random_simulation/pipa_simulation_SL/Sample_ID_random
``` 
For each species, after the simulations, the input pedigree files will be compress to save space using:
```
tar -czvf pipa_sampleID_1000simulations.tar.gz /4/caroline/random_simulation/pipa_simulation_SL/Sample_ID_random*
``` 

### Notes
**Sept.20:** Meeting with JD to discuss about the tests and make sure it makes sense. Additional notes will be posted here after the meeting.

*Main idea*: using a hypergeometric distribution to assign a score to the site; then do a permutation test (comparing the value obtain with the data compared to random).

Comment from BE (25/09): A non-significant result can just be due to the small size of the sex-linked region. Need to keep that in mind in the discussion.

Some comments on [Pigneau et al. 2004](https://academic.oup.com/mbe/article-lookup/doi/10.1093/molbev/msh244):

- measured the relationship between measure of linkage desequilibrium r^2 and distance between sites (Mantel test): we can't really do it since we have a *de novo* approach and a lot of scaffolds.
- ~1000 randomizations (number depends on the test)
- to combine results across datasets: Fisher's method of combining probabilities by `\sum -2Ln(p)`
```
Z = -2 { (\sum_{i=1}^{nsex} Ln(p_i)/nsex) - (\sum_{i=1}^{nasex} Ln(p_i)/nasex)}
```
Should we compare the results between our 2 species? with for us `n=number of sites for each species`?

**4/10:** Random ID files copied on Brian machine to test next week, to save space they have been archive for now and will be only uncompress during the run.
```
tar -czvf random_ID_Hymenochirus.tar.gz /4/caroline/random_simulation/random_ID_Hymenochirus
```
The 1st part script have been updated and now contains the functions to obtain the scores. Need to add the scores to the output and using it in the `c++` script. 

### Readings 

- [Lunetta et al. 200](http://www.sciencedirect.com/science/article/pii/S000292970763434X)

- [Palaiokostas et al. 2013](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-14-566)

- [Yurii et al. 2007](http://www.genetics.org/content/177/1/577)

